---
title: "TFM2_Taraspina_miTags_analysis"
author: "lrubinat"
date: "5/23/2017"
output:
  html_document:
    theme: united
    toc: yes
  pdf_document:
    highlight: zenburn
    toc: yes
---


<!--- INITIALIZATION
```{r, echo=FALSE}
#error hook to kill knitr in case of errors
library(knitr)
knit_hooks$set(error = function(x, options) stop(x))
opts_chunk$set(cache=TRUE, autodep=TRUE)
```
--->

```{r libraries, echo=F, results="hide", warning=F, message=F}
library(data.table)
library(dplyr)
library(DT)
library(ggplot2)
library(gtools)
library(vegan)
library(iNEXT)
library(fossil)
library(ggrepel)
library(dendextend)
library(forcats)
```



# 1) 18S amplicons

## 1.1) Data overview


``` {r load_data, echo=FALSE, message=FALSE, results="hide", warning=F}
#Let's read the dataset and remove the samples containing less than 1154 reads:

setwd("~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/")

#read data 
tb18_tax <- read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/otu_table99_SUR_miTags_18S_classif.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(tb18_tax) # 8642   58
tb18_tax[1:5,1:5]

row.names(tb18_tax)<-tb18_tax[,1]
tb18_tax<-tb18_tax[,-1]
tb18_tax[is.na(tb18_tax)]<-"NA"

dim(tb18_tax) #8642   57
tb18_tax[1:5,1:5]

#table with taxonomic classification alone:
tb18_class <- tb18_tax[,56:57]
dim(tb18_class) #8642    2
tb18_class[1:5,1:2]

#table with occurence data alone
tb18_tax_occur <- tb18_tax[,1:55]
dim(tb18_tax_occur) #8642   55
tb18_tax_occur[1:5,1:5]
```

The dataset "Taraspina 18S miTags" contains reads from **45 samples of Tara Oceans** and from **10 samples of Malaspina**. On average, each sample contains 4440 OTUs:

```{r summary, echo=FALSE, message=FALSE, warning=F}
amplicons_per_sample_tb18<-colSums(tb18_tax_occur)
summary(amplicons_per_sample_tb18) 
#median of amplicons per sample = ~3086)
```

```{r samples_selection, echo=FALSE, message=FALSE, results="hide", warning=F}
#we'll remove all samples with less than 1154 reads.
amplicons_per_sample_tb18[which(colSums(tb18_tax_occur)<1154)]

#remove samples with less than 1154 reads
tb18_tax_occur_min1154 <- tb18_tax_occur[,colSums(tb18_tax_occur) >= 1154]
dim(tb18_tax_occur_min1154) #8642   45

#remove samples with omitted in 16S dataset (so that we can compare the relative abundance of 16S and 18S OTUs considering the same samples)
tb18_tax_occur_min1154<-subset(tb18_tax_occur_min1154, select=-c(TARA_70_SUR_0d2_3,TARA_67_SUR_0d2_3))


dim(tb18_tax_occur_min1154) #8642   43

#let's check if we loose any OTU when excluding the mentioned stations.
tb18_tax_occur_min1154<-tb18_tax_occur_min1154[-(which(rowSums(tb18_tax_occur_min1154)==0)),]
dim(tb18_tax_occur_min1154)
```

```{r starting_dataset, echo=FALSE, results="hide"}
#Table dimensions and content outline:

dim(tb18_tax_occur_min1154)
#tb18_tax_occur_min1154[1:5,1:5]
```

```{r reads_per_sample_overview1, echo=F, results="hide"}
#Minimum number of reads per station:

min(colSums(tb18_tax_occur_min1154)) 
#1154
```

```{r reads_per_sample_overview2, echo=F, results="hide"}
#Maximum number of reads per station:

max(colSums(tb18_tax_occur_min1154))
# max: 23781
```

```{r reads_per_sample_overview3, echo=F, results="hide"}
#Identification of stations with higher number of reads:

amplicons_per_sample<-colSums(tb18_tax_occur_min1154)
amplicons_per_sample[which(colSums(tb18_tax_occur_min1154)>20000)]
```
\
Overall reads per sample:

``` {r reads_per_sample_overview4, echo=FALSE}

plot(sort(colSums(tb18_tax_occur_min1154)), pch=19, xlab="sample no.", ylab="reads per sample", cex=0.9)
```


## 1.2) Normalization

\
In order to keep as many samples as possible, we rarefy at **1154 reads per sample**. By that, we loose 10 samples, and after removing the exluded samples in the 16S dataset (to make them comparable), we end up with a **normalized dataset containing 43 samples and 5669 OTUs**.


``` {r species_richness_rarefaction1, echo=FALSE, results="hide"}
#Let's normalize the original dataset by randomly subsampling 1154 reads in each station:

tb18_tax_occur_min1154_t<-t(tb18_tax_occur_min1154)
tb18_tax_occur_ss1154<-rrarefy(tb18_tax_occur_min1154_t, 1154)
```

```{r species_richness_rarefaction2, echo=FALSE, results="hide"}
dim(tb18_tax_occur_ss1154)
#tb18_tax_occur_ss1154[1:5,1:5]
```

``` {r species_richness_rarefaction3, echo=F, results="hide"}
#Its content fits with the expected normalization values (1154 reads per station):

rowSums(tb18_tax_occur_ss1154)
```

```{r species_richness_rarefaction4, echo=F, results="hide"}
#OTUs we loose in the new table:

length(which(colSums(tb18_tax_occur_ss1154)==0))
```

```{r species_richness_rarefaction5, echo=F, results="hide"}
#let's remove them from the table and take a look at its final dimensions:

tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154[,-(which(colSums(tb18_tax_occur_ss1154)==0))]
tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154_no_cero[mixedorder(row.names(tb18_tax_occur_ss1154_no_cero)),]
dim(tb18_tax_occur_ss1154_no_cero)

#the final dimensions of the normalized table are 43 5669.
#5669 + 2973 = 8642
```
\
Datasets summary:
```{r dataset_summary,  echo=T}
dim(tb18_tax) #original dataset

dim(tb18_tax_occur) #original dataset with occurrence data alone

dim(tb18_tax_occur_min1154) #dataset without samples with less than 1154 OTUs

dim(tb18_tax_occur_ss1154_no_cero) #rarefied dataset
```



## 1.3) General community analysis

### 1.3.1) Richness and evenness (Shannon index)

```{r shannon_index1, echo=FALSE}
tb18_tax_occur_ss1154_div <- diversity(tb18_tax_occur_ss1154_no_cero, index="shannon")
```

Most of the samples take Shannon Index values around 6:

```{r shannon_index2, echo=FALSE}
boxplot(tb18_tax_occur_ss1154_div, pch=19, main="TP 18S Shannon's index of diversity",ylab="Shannon's index")
plot(sort(tb18_tax_occur_ss1154_div), pch=19, main="TP 18S Shannon's index of diversity", xlab="sample no.", ylab="Shannon's index")
```

### 1.3.2) Richness: OTU number

```{r richness_otu_no1, echo=FALSE}
OTUs_per_sample_18S_tax_occur_ss1154<-specnumber(tb18_tax_occur_ss1154_no_cero)
```

Lowest number of OTUs per sample:

```{r richness_otu_no2, echo=FALSE}
min(OTUs_per_sample_18S_tax_occur_ss1154)
```

Maximum number of OTUs per sample:

```{r richness_otu_no3, echo=FALSE}
max(OTUs_per_sample_18S_tax_occur_ss1154)
```

In most of the samples, we can identify between 600 and 700 OTUs:

```{r richness_otu_no4, echo=F}
plot(sort(OTUs_per_sample_18S_tax_occur_ss1154), pch=19, main="TP 18S no. of OTUs per sample", xlab="sample no.", ylab="no. of OTUs")
boxplot(OTUs_per_sample_18S_tax_occur_ss1154, pch=19, main="TP 18S no. of OTUs per sample", ylab="no. of OTUs")
```

### 1.3.3) Index of evenness

#### 1.3.3.1) Pielou's index

```{r pielou_index_of_evenness1, echo=F, results="hide"}
pielou_evenness_18S_tax_occur_ss1154<-tb18_tax_occur_ss1154_div/log(OTUs_per_sample_18S_tax_occur_ss1154)
```

The Pielou index (constrained between 0 and 1) takes values closer to 1 as the variation of species proportion in a sample increases. Most of the samples get values between 0.90 and 0.95, meaning that the numerical composition of different OTUs in a sample is highly variable - there's no constant presence of dominant species.

```{r pielou_index_of_evenness2, echo=F}
plot(sort(pielou_evenness_18S_tax_occur_ss1154), pch=19, main="TP 18S Pielou's index", xlab="sample no.", ylab="Pielou's index")
boxplot(pielou_evenness_18S_tax_occur_ss1154, pch=19,  main="TP 18S Pielou's index", ylab="Pielou's index")
```


<!---
#### 1.3.3.2) Sads
library(sads)
?sads
--->

### 1.3.4) Abundance Models


```{r OTUs_overall_abundance, echo=F, results="hide"}
#The OTU_38, with 874 reads, is the most abundant in the overall dataset:

head(sort(colSums(tb18_tax_occur_ss1154_no_cero), decreasing=T), n=10L)
```

Most of the OTUs show very few occurrences, suggesting that we will probably be able to identify a significant ammount of rare otus:

```{r OTUs_overall_abundance2, echo=F, results="hide"}
#previous version:
#plot(log(sort(colSums(tb18_tax_occur_ss1154_no_cero), decreasing=T)), pch=19)

tb18_tax_occur_ss1154_no_cero_rank<-as.data.table(t(tb18_tax_occur_ss1154_no_cero))
tb18_tax_occur_ss1154_no_cero_rank[,V1:=rowSums(tb18_tax_occur_ss1154_no_cero_rank)]
tb18_tax_occur_ss1154_no_cero_rank<-tb18_tax_occur_ss1154_no_cero_rank[order(V1,decreasing = T)]
tb18_tax_occur_ss1154_no_cero_rank[,rank:=1:nrow(tb18_tax_occur_ss1154_no_cero_rank)]

p1<-ggplot(tb18_tax_occur_ss1154_no_cero_rank, aes(x=rank,y=V1))+
  geom_point(aes(x=rank,y=V1))+
  scale_y_log10(name="abundance (log scale)")+
  labs(title="TP 18S rank abundance curve")+
  theme_bw()+
  labs(colour = "") 
p1
```



#### 1.3.4.1) Rank-Abundance or Dominance/Diversity Model ("radfit")

The OTUs abundance distribution fits relativelly close to log-normal model. 

```{r radfit, echo=FALSE}
#?radfit
#otu_tb18_t[1:5,1:5]

tb18_tax_occur_min1154_radfit<-radfit(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_min1154_radfit)
```

#### 1.3.4.2) Preston's Log-normal Model

According to Preston's lognormal model fitted into groups of species' frequencies, we're missing 1588.816 species:

```{r preston_model1, echo=F}
tb18_tax_occur_ss1154_prestonfit<-prestonfit(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_ss1154_prestonfit, main="TP 18S fit to Preston log-normal model \n(pooled species)")

veiledspec(tb18_tax_occur_ss1154_prestonfit)
```

\
When computing Prestons' lognormal model fit without pooling data into groups, we miss 1385.371 species:

```{r preston_model2, echo=4}
tb18_tax_occur_ss1154_dist_all<-prestondistr(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_ss1154_prestonfit, main="TP 18S fit to Preston log-normal model \n(red: pooled; blue: non-pooled)")
lines(tb18_tax_occur_ss1154_dist_all, line.col="blue3")

veiledspec(tb18_tax_occur_ss1154_dist_all)
```


### 1.3.5) Rarefaction curves of rarefied and non-rarefied datasets

```{r rarefraction_curve, echo=FALSE}
rarec_input<-t(as.matrix(colSums(tb18_tax_occur_ss1154_no_cero)))

#tb18_rarecurve_step1000_10000<-rarecurve(rarec_input, step = 1000, 10000, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity step=1000 & ss=10000\n(40,816 OTUs; 5,247,375 reads)\n")
#tb18_rarecurve_step1000_10000

tb18_rarecurve_step1000_ss1154<-rarecurve(rarec_input, step = 100, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity, step=100 \n(5,719 OTUs; 49,622 reads)\n")


#without rarefying
rarec_allOTUs_input<-t(as.matrix(colSums(t(tb18_tax_occur))))
tb18_rarecurve_allOTUs_step1000_ss8642<-rarecurve(rarec_allOTUs_input, step = 100, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity non-rarefied, step=100 \n(8,642 OTUs; 244,184 reads)\n")
```


### 1.3.6) Beta diversity

#### 1.3.6.1) Dissimilarity matrix using Bray-Curtis index:

The Bray-Curtis dissimilarity, constrained between 0 (minimum distance) and 1 (highest dissimilarity) allows us to quantify the differences between samples according to the composition and the relative abundance of their OTUs. In our dataset, most of the samples pairs take dissimilarity values around 0.8, meaning that their composition is substantially different.

```{r beta_div1, echo=FALSE}
#?vegdist
tb18_tax_occur_ss1154_no_cero.bray<-vegdist(tb18_tax_occur_ss1154_no_cero, method="bray")
boxplot(tb18_tax_occur_ss1154_no_cero.bray, main="Bray-Curtis dissimilarity matrix")
```

#### 1.3.6.2) Hierarchical clustering

The stations seem to form clusters according to geographic localization, but there are no evident clusters separated from the general groups, apart from the one containing TARA_84, TARA_82 and TARA_85.

(To be done: assign Longhurst provinces information to each station and check if any of the central clusters is meaningful regarding to the samples' geographical location)

```{r beta_div2, echo=FALSE, results="hide"}
#UPGMA
tb18_tax_occur_ss1154_no_cero.upgma<-hclust(tb18_tax_occur_ss1154_no_cero.bray, "average")
plot(tb18_tax_occur_ss1154_no_cero.upgma, cex=.35, main="Samples Hierarchical Clustering")
```


```{r hclust_coloured, echo=FALSE}
#we'll repeat the UPGMA with the Longhurst codes added in the stations name
longhurst<-read.table("input/TP_longhurst_provinces_merged.txt",header=F,sep="\t")
row.names(longhurst)<-longhurst$V1
tb18_tax_occur_ss1154_no_cero_dend<-merge(tb18_tax_occur_ss1154_no_cero, longhurst, by="row.names", all.x=TRUE)
row.names(tb18_tax_occur_ss1154_no_cero_dend)<-tb18_tax_occur_ss1154_no_cero_dend$V3
tb18_tax_occur_ss1154_no_cero_dend<-tb18_tax_occur_ss1154_no_cero_dend[ , -which(names(tb18_tax_occur_ss1154_no_cero_dend) %in% c("Row.names","V1","V2","V3"))]

tb18_tax_occur_ss1154_no_cero_dend.bray<-vegdist(tb18_tax_occur_ss1154_no_cero_dend, method="bray")
tb18_tax_occur_ss1154_no_cero_dend.upgma<-hclust(tb18_tax_occur_ss1154_no_cero_dend.bray, "average")
plot(tb18_tax_occur_ss1154_no_cero_dend.upgma, cex=.35, main="Samples Hierarchical Clustering")
```

<!--
```{r hclust_coloured2, echo=F, results="hide"}
dend <- as.dendrogram(tb18_tax_occur_ss1154_no_cero_dend.upgma)

###########

# Create a vector giving a color for each province
provinces <- rep("Other", length(rownames(tb18_tax_occur_ss1154_no_cero_dend)))
is_x <- grepl("SATL", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "SATL"
is_x <- grepl("SPSG", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "SPSG"
is_x <- grepl("PEQD", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "PEQD"
is_x <- grepl("NASE", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "NASE"
is_x <- grepl("SPO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "SPO"
is_x <- grepl("NPO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "NPO"
is_x <- grepl("NAO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "NAO"
is_x <- grepl("IO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "IO"
is_x <- grepl("SAO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "SAO"
is_x <- grepl("SO", rownames(tb18_tax_occur_ss1154_no_cero_dend))
provinces[is_x] <- "SO"

provinces <- factor(provinces)
n_provincess <- length(unique(provinces))
cols_4 <- colorspace::rainbow_hcl(n_provincess, c = 70, l  = 50)
col_provinces <- cols_4[provinces]

k234 <- cutree(dend, k = 2:4)

labels_colors(dend) <- col_provinces[order.dendrogram(dend)]
dend <- color_branches(dend, k = 4)

par(mar = c(10,4,2,12))
plot(dend, horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_provinces), dend, rowLabels = c(paste0("k = ", 4:2), "Provinces"), horiz = TRUE)
legend(1,-8, legend=c("Indic", "N. Atlantic", "N. Atlantic Subtropical Gyral Province (East)", "N. Pacific Ocean", "Pacific Equatorial Divergence", "S. Atlantic", "S. Atlantic", "Southern Ocean", "S. Pacific", "S. Pacific Subtropical Gyre"), fill = cols_4, cex=0.8, ncol = 3)

#legend(1, 95, legend=c("Line 1", "Line 2"),
#       col=c("red", "blue"), lty=1:2, cex=0.8)
dev.off()
```
-->

#### 1.3.6.3) Non-metric multidimensional scaling

We can identify a prominent group in the central part of the NMDS plot and a few outliers (TARA 82, 84 and 85) in the central-right edge of the plot. The stress parameter takes a value below 0.2, suggesting that the plot is acceptable. 

```{r monoNMDS, echo=F}
#NMDS
tb18_tax_occur_ss1154_no_cero.nmds<-monoMDS(tb18_tax_occur_ss1154_no_cero.bray)
tb18_tax_occur_ss1154_no_cero.nmds
plot(tb18_tax_occur_ss1154_no_cero.nmds, main="monoMDs method")
```


```{r metaNMDS, echo=F, results="hide", warning=F}
#When implementing a most robut function for computing NMDS plots, the result is quiet the same:

tb18_tax_occur_ss1154_no_cero.meta_nmds<-metaMDS(tb18_tax_occur_ss1154_no_cero.bray)
plot(tb18_tax_occur_ss1154_no_cero.meta_nmds, main="metaMDS method")
```

## 1.4) Geographical analysis

```{r load_geo_data, echo=F, results="hide", message=F, warning=F}
#load geographical ubication of stations and sort according to otu_tb18 stations sequence.
MP_geo_18S<-read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/Taraspina_SUR_loc_nodot_stand.name2.txt", sep="\t", header=T)

MP_geo_18S<-data.table(MP_geo_18S)
#MP_geo_18S[,st_name:=(paste("st",MP_geo_18S$station,"_MD",MP_geo_18S$code,sep=""))]
MP_geo_18S<-MP_geo_18S[mixedorder(MP_geo_18S$sample),]
tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154_no_cero[order(row.names(tb18_tax_occur_ss1154_no_cero)),]
#select only stations apearing in tb18_tax_occur_ss1154_no_cero:
MP_geo_18S<-MP_geo_18S[MP_geo_18S$sample %in% row.names(tb18_tax_occur_ss1154_no_cero),]
row.names(MP_geo_18S)<-MP_geo_18S$sample

dim(MP_geo_18S)
MP_geo_18S[1:5,1:3]
tb18_tax_occur_ss1154_no_cero[1:5,1:5]

#we'll use a function from library(fossil) to read lat-long in decimal degrees and translate into distance in km. We'll print only columns containing info about station, latitude and longitude:
MP_geo_18S_reduced<-create.lats(MP_geo_18S, loc="sample", long="long", lat="lat")
head(MP_geo_18S_reduced)

#create a distance matrix (lower triangle) between a list of points.
geo_distances_MP_18S<-earth.dist(MP_geo_18S_reduced, dist = TRUE)
head(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

geo_distances_MP_18S<-as.matrix(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

#geo distances dataset ready to use: "geo_distances_MP_18S"
```


```{r working_datasets1, echo=F, results="hide", warning=F}
#Working datasets:

#1) Community matrix: tb18_tax_occur_ss1154_no_cero
dim(tb18_tax_occur_ss1154_no_cero)
tb18_tax_occur_ss1154_no_cero[1:5, 1:5]

#2) Community Bray-Curtis: tb18_tax_occur_ss1154_no_cero.bray
tb18_tax_occur_ss1154_no_cero.bray<-vegdist(tb18_tax_occur_ss1154_no_cero, method="bray")
tb18_tax_occur_ss1154_no_cero.bray<-as.matrix(tb18_tax_occur_ss1154_no_cero.bray)
dim(tb18_tax_occur_ss1154_no_cero.bray)

#3) Stations distances in km: geo_distances_MP_18S
dim(geo_distances_MP_18S)
```

Communities quickly change their composition across geographical distances:

```{r working_datasets4, echo=F}
plot(geo_distances_MP_18S, tb18_tax_occur_ss1154_no_cero.bray, pch=19, cex=0.4, xlab="Geopgraphical distances", ylab="Bray-Curtis dissimilarities")
```

### 1.4.1) Mantel correlograms

Mantel statistic is -significantlly- so low, meaning that the correlation between samples dissimilarity and geographical distances is weak.

```{r mantel_correlogram1, echo=F}
mantel(geo_distances_MP_18S, tb18_tax_occur_ss1154_no_cero.bray)
```


```{r mantel_correlogram2, echo=F, results="hide"}
#Maximum distance between samples:
max(geo_distances_MP_18S)

#Minimum distance between samples:
min(geo_distances_MP_18S)
```

Correlograms:

```{r mantel_correlogram4, echo=T}
MP_18s_ss1154_mantel_correl_by_1000km<-mantel.correlog(tb18_tax_occur_ss1154_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=1000))
plot(MP_18s_ss1154_mantel_correl_by_1000km)

MP_18s_ss1154_mantel_correl_by_100km<-mantel.correlog(tb18_tax_occur_ss1154_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=100))
plot(MP_18s_ss1154_mantel_correl_by_100km)
```

## 1.5) Abundance vs. occurence

```{r OTUs_mean_relative_abund, echo=F, results="hide"}
tb18_tax_occur_ss1154_no_cero[1:5,1:5]
tb18_tax_occur_ss1154_no_cero_t<-t(tb18_tax_occur_ss1154_no_cero)

colSums(tb18_tax_occur_ss1154_no_cero_t)

#local abundance percentage
tb18_tax_occur_ss1154_no_cero_t.rabund<-tb18_tax_occur_ss1154_no_cero_t/1154

colSums(tb18_tax_occur_ss1154_no_cero_t.rabund)
tb18_tax_occur_ss1154_no_cero_t.rabund[1:5,1:5]

#OTUs mean relative abundance
tb18_tax_occur_ss1154_no_cero_t.rabund_means<-rowMeans(tb18_tax_occur_ss1154_no_cero_t.rabund) 
tb18_tax_occur_ss1154_no_cero_t.rabund_means<-as.data.frame(tb18_tax_occur_ss1154_no_cero_t.rabund_means)

head(tb18_tax_occur_ss1154_no_cero_t.rabund_means)
```

```{r OTUs_occurence, echo=F, results='hide'}
tb18_tax_occur_ss1154_no_cero_t.rabund.occur<-tb18_tax_occur_ss1154_no_cero_t.rabund
tb18_tax_occur_ss1154_no_cero_t.rabund.occur[tb18_tax_occur_ss1154_no_cero_t.rabund.occur>0]<-1
tb18_tax_occur_ss1154_no_cero_t.rabund.occur[1:5,1:5] # presence - absence table

#percentage of occurence in overall stations
tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc<-as.data.frame(100*(rowSums(tb18_tax_occur_ss1154_no_cero_t.rabund.occur)/43))

str(tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc)
```

```{r merge_rabund_peroccur, echo=F, results='hide'}
otu_tb18_ss1154_rabund_percoccur<-merge(tb18_tax_occur_ss1154_no_cero_t.rabund_means,tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc, by="row.names")

colnames(otu_tb18_ss1154_rabund_percoccur)<-c("OTUs","mean_rabund","perc_occur")
otu_tb18_ss1154_rabund_percoccur[1:5,]

row.names(otu_tb18_ss1154_rabund_percoccur)<-otu_tb18_ss1154_rabund_percoccur[,1]
otu_tb18_ss1154_rabund_percoccur<-otu_tb18_ss1154_rabund_percoccur[,-1]
otu_tb18_ss1154_rabund_percoccur[1:5,]
```

OTUs distribution according to their percentage of occurence and relative abundance.\
- red line: OTUs that occur in more than 80% of the samples.\
- blue line: regionally abundant OTUs (> 0.1%).

```{r abund_vs_occurence_table, echo=F}
plot(otu_tb18_ss1154_rabund_percoccur$mean_rabund,otu_tb18_ss1154_rabund_percoccur$perc_occur, log="x", pch=19, cex=0.8, xlab="Mean relative abundance", ylab="Percentage of occurence")
abline(h=80, col="red") #occurence higher than 80%
abline(v=0.00001, col="green") #rare OTUs
abline(v=0.001, col="blue") #cosmopolitan OTUs

#Conventional limits:
#Regionally rare     = 0.00001
#Regionally abundant = 0.001
```

Regionally abundant OTUs (relative abundance over 0.1%):

```{r abundant_OTUs, echo=F}
#regionally abundant
tb18_ss1154_abundant<-otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund > 0.001,]

tb18_ss1154_abundant_sorted<-tb18_ss1154_abundant[order(tb18_ss1154_abundant$mean_rabund, tb18_ss1154_abundant$perc_occur, decreasing = T), c(1,2)]

tb18_ss1154_abundant_sorted_prov<-cbind(otu_names=row.names(tb18_ss1154_abundant_sorted),tb18_ss1154_abundant_sorted)
tb18_class_prov<-cbind(otu_names=row.names(tb18_class),tb18_class)
#tb18_class_prov<-tb18_class_prov[,-2]
tb18_ss1154_abundant_sorted_prov<-merge(tb18_ss1154_abundant_sorted_prov,tb18_class_prov, by="otu_names", all.x=TRUE)

tb18_ss1154_abundant_sorted_prov<-tb18_ss1154_abundant_sorted_prov[order( tb18_ss1154_abundant_sorted_prov$mean_rabund, tb18_ss1154_abundant_sorted_prov$perc_occur, decreasing = T), c(1,2,3,5,4)]
tb18_ss1154_abundant_sorted_prov
#dim(tb18_ss1154_abundant_sorted_prov)
```

Number and roportion of regionally abundant OTUs (%):

```{r abundant_OTUs2, echo=F}
nrow(tb18_ss1154_abundant_sorted_prov)

(nrow(tb18_ss1154_abundant_sorted_prov)/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```

\
Cosmopolitan OTUs (relative abundance over 0.1% and occurence in more than 80% of samples):

```{r select_cosmopolitan, echo=F}
otu_tb18_ss1154_rabund_cosm<-otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund > 0.001,]
otu_tb18_ss1154_rabund_poccur_cosm<-otu_tb18_ss1154_rabund_cosm[otu_tb18_ss1154_rabund_cosm$perc_occur > 80,]
otu_tb18_ss1154_cosmop_sorted<-otu_tb18_ss1154_rabund_poccur_cosm[order(otu_tb18_ss1154_rabund_poccur_cosm$perc_occur, otu_tb18_ss1154_rabund_poccur_cosm$mean_rabund, decreasing = T), c(1,2)]


otu_tb18_ss1154_cosmop_sorted_prov<-cbind(otu_names=row.names(otu_tb18_ss1154_cosmop_sorted),otu_tb18_ss1154_cosmop_sorted)
otu_tb18_ss1154_cosmop_sorted_prov<-merge(otu_tb18_ss1154_cosmop_sorted_prov,tb18_class_prov, by="otu_names", all.x=TRUE)

otu_tb18_ss1154_cosmop_sorted_prov<-otu_tb18_ss1154_cosmop_sorted_prov[order(otu_tb18_ss1154_cosmop_sorted_prov$perc_occur, otu_tb18_ss1154_cosmop_sorted_prov$mean_rabund, decreasing = T), c(1,2,3,5,4)]
otu_tb18_ss1154_cosmop_sorted_prov
#dim(otu_tb18_ss1154_cosmop_sorted_prov)
```

Number and proportion (%) of cosmopolitan OTUs:

```{r percentage_cosmopolitan, echo=F}
nrow(otu_tb18_ss1154_cosmop_sorted_prov)

(nrow(otu_tb18_ss1154_cosmop_sorted_prov)/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```

Number and proportion (%) of rare OTUs:

```{r rare_OTUs, echo=F}
nrow(otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund < 0.00001 & otu_tb18_ss1154_rabund_percoccur$mean_rabund >0,])

(nrow(otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund < 0.00001 & otu_tb18_ss1154_rabund_percoccur$mean_rabund >0,])/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```




## 1.6) Taxonomic composition analysis

### 1.6.1) Normalized data

\
**PHOTOTROPHS + HETEROTROPHS**


```{r merge_tables, echo=FALSE, results="hide", warning=FALSE}
#Let's add the taxonomic classification by merging "tb18_tax_occur_ss1154_no_cero" with "tb18_tax":

tb18_tax_occur_ss1154_no_cero_t<-t(tb18_tax_occur_ss1154_no_cero)
tb18_ss1154_tax<-merge(tb18_tax_occur_ss1154_no_cero_t,tb18_class, by="row.names", all.x=T)

dim(tb18_ss1154_tax)
tb18_ss1154_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_ss1154_tax)=tb18_ss1154_tax$Row.names
#add OTU_no as rowname
rownames.tb18_ss1154_tax<-tb18_ss1154_tax[,1]
tb18_ss1154_tax<-tb18_ss1154_tax[,-1]

dim(tb18_ss1154_tax)
tb18_ss1154_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_ss1154_tax["OTU_no"] <- NA
tb18_ss1154_tax$OTU_no <- sapply(strsplit(rownames(tb18_ss1154_tax),split= "\\_"),'[',2)
tb18_ss1154_tax$OTU_no <- as.numeric(as.character(tb18_ss1154_tax$OTU_no))
tb18_ss1154_tax_sorted<-tb18_ss1154_tax[order(tb18_ss1154_tax$OTU_no, decreasing = FALSE), ]
```

No. of OTUs and reads of the rearefied dataset:

```{r dim_merge1, echo=F}
nrow(tb18_ss1154_tax_sorted)
sum(colSums(tb18_ss1154_tax_sorted[,1:43]))

#dim(tb18_ss1154_tax_sorted)
#tb18_ss1154_tax_sorted[1:5,1:5]
```

No. of OTUs and reads of phototrophic groups:

```{r dim_merge2, echo=F}
#selection of phototrophs:
tb18_phototrophs <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif != "NA"),]

nrow(tb18_phototrophs)
sum(colSums(tb18_phototrophs[,1:43]))

#dim(tb18_phototrophs)
#tb18_phototrophs[1:5,1:5]
```


No. of OTUs and reads of non-phototrophic groups:

```{r dim_merge3, echo=F}
#translate "NA" to "Heterotrophs":
tb18_ss1154_tax_sorted$classif<-as.character(tb18_ss1154_tax_sorted$classif)
tb18_ss1154_tax_sorted$classif[which(is.na(tb18_ss1154_tax_sorted$classif))]<-"Heterotrophs"

#selection of non.phototrophs:
tb18_non.phototrophs <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Heterotrophs"),]

nrow(tb18_non.phototrophs)
sum(colSums(tb18_non.phototrophs[,1:43]))

#dim(tb18_non.phototrophs)
#tb18_non.phototrophs[1:5,1:5]
```


\
**Absolute values**


```{r count_samples_per_group_Heterotrophs, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

Heterotrophs_tb <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif == "Heterotrophs"),]
Heterotrophs_tb_occur <- Heterotrophs_tb[,1:43]
Heterotrophs_tb_occur_len<-length(Heterotrophs_tb_occur[,colSums(Heterotrophs_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Heterotrophs",samples_per_class=Heterotrophs_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

```{r aggregate_Heterotrophs, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_ss1154_tax_sorted[1:43]), list(tb18_ss1154_tax_sorted$classif), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_ss1154_tax_sorted[1:43]), list(tb18_ss1154_tax_sorted$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_Heterotrophs, echo=FALSE, warning=F}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples<-merge(tb18S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]
tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
tb18S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S_Heterotrophs, echo=FALSE, warning=F}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2_Heterotrophs, echo=FALSE, warnings=F, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-18S OTUs (%)", y="TP-miTags-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S_Heterotrophs, echo=FALSE, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-18S samples (%)", y="TP-miTags-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
**PHOTOTROPHS**

```{r count_samples_per_group, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

\
**Absolute values**

```{r aggregate, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples, echo=FALSE, warning=F}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples<-merge(tb18S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]
tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
tb18S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S, echo=FALSE, warning=F}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2, echo=FALSE, warnings=F, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-18S OTUs (%)", y="TP-miTags-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S, echo=FALSE, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-18S samples (%)", y="TP-miTags-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```



### 1.6.2) Non-rarefied data

\
**PHOTOTROPHS + HETEROTROPHS**


```{r merge_tables_non.raref, echo=FALSE, results="hide", warning=FALSE}
#Let's add the taxonomic classification by merging "tb18_tax_occur_min1154_no_cero" with "tb18_tax":

tb18_tax_occur_min1154_t<-tb18_tax_occur_min1154
tb18_min1154_tax<-merge(tb18_tax_occur_min1154_t,tb18_class, by="row.names")


dim(tb18_min1154_tax)
tb18_min1154_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_min1154_tax)=tb18_min1154_tax$Row.names
#add OTU_no as rowname
rownames.tb18_min1154_tax<-tb18_min1154_tax[,1]
tb18_min1154_tax<-tb18_min1154_tax[,-1]

dim(tb18_min1154_tax)
tb18_min1154_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_min1154_tax["OTU_no"] <- NA
tb18_min1154_tax$OTU_no <- sapply(strsplit(rownames(tb18_min1154_tax),split= "\\_"),'[',2)
tb18_min1154_tax$OTU_no <- as.numeric(as.character(tb18_min1154_tax$OTU_no))
tb18_min1154_tax_sorted<-tb18_min1154_tax[order(tb18_min1154_tax$OTU_no, decreasing = FALSE), ]
```

No. of OTUs and reads of the rearefied dataset:

```{r dim_merge1_non.raref, echo=F}
nrow(tb18_min1154_tax_sorted)
sum(colSums(tb18_min1154_tax_sorted[,1:43]))

#dim(tb18_min1154_tax_sorted)
#tb18_min1154_tax_sorted[1:5,1:5]
```

No. of OTUs and reads of phototrophic groups:

```{r dim_merge2_non.raref, echo=F}
#selection of phototrophs:
tb18_phototrophs <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif != "NA"),]

nrow(tb18_phototrophs)
sum(colSums(tb18_phototrophs[,1:43]))

#dim(tb18_phototrophs)
#tb18_phototrophs[1:5,1:5]
```


No. of OTUs and reads of non-phototrophic groups:

```{r dim_merge3_non.raref, echo=F}
#translate "NA" to "Heterotrophs":
tb18_min1154_tax_sorted$classif<-as.character(tb18_min1154_tax_sorted$classif)
tb18_min1154_tax_sorted$classif[which(is.na(tb18_min1154_tax_sorted$classif))]<-"Heterotrophs"

#selection of non.phototrophs:
tb18_non.phototrophs <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Heterotrophs"),]

nrow(tb18_non.phototrophs)
sum(colSums(tb18_non.phototrophs[,1:43]))

#dim(tb18_non.phototrophs)
#tb18_non.phototrophs[1:5,1:5]
```


\
**Absolute values**


```{r count_samples_per_group_Heterotrophs_non.raref, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

Heterotrophs_tb <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif == "Heterotrophs"),]
Heterotrophs_tb_occur <- Heterotrophs_tb[,1:43]
Heterotrophs_tb_occur_len<-length(Heterotrophs_tb_occur[,colSums(Heterotrophs_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Heterotrophs",samples_per_class=Heterotrophs_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

```{r aggregate_Heterotrophs_non.raref, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_min1154_tax_sorted[1:43]), list(tb18_min1154_tax_sorted$classif), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_min1154_tax_sorted[1:43]), list(tb18_min1154_tax_sorted$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_Heterotrophs_non.raref, echo=FALSE, warning=F}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples<-merge(tb18S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]
tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
tb18S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S_Heterotrophs_non.raref, echo=FALSE, warning=F}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2_Heterotrophs_non.raref, echo=FALSE, warnings=F, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-18S OTUs (%)", y="TP-miTags-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S_Heterotrophs_non.raref, echo=FALSE, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-18S samples (%)", y="TP-miTags-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
**PHOTOTROPHS**

```{r count_samples_per_group_non.raref, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

\
**Absolute values**

```{r aggregate_non.raref, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_non.raref, echo=FALSE, warning=F}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples<-merge(tb18S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]
tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
tb18S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S_non.raref, echo=FALSE, warning=F}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2_non.raref, echo=FALSE, warnings=F, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-18S OTUs (%)", y="TP-miTags-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S_non.raref, echo=FALSE, warning=F}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-18S samples (%)", y="TP-miTags-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


\
\
\
\


# 2) 16S amplicons

## 2.1) Data overview


``` {r load_data_16S, echo=FALSE, message=FALSE, results="hide", warning=F}
#Let's read the dataset and remove the samples containing less than 36155 reads:

#read data 
tb16_tax <- read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/otu_table99_SUR_miTags_16S_classif.txt", head=TRUE, fill=TRUE, sep="\t")

#table dimensions and format before setting column names
dim(tb16_tax) # 28294    59
tb16_tax[1:5,1:5]

row.names(tb16_tax)<-tb16_tax[,1]
tb16_tax<-tb16_tax[,-1]
tb16_tax[is.na(tb16_tax)]<-"NA"

dim(tb16_tax) #28294    58
tb16_tax[1:5,1:5]

#table with taxonomic classification alone:
tb16_class <- tb16_tax[,56:58]
dim(tb16_class) #28294     3
tb16_class[1:5,1:3]

#table with occurence data alone
tb16_tax_occur <- tb16_tax[,1:55]
dim(tb16_tax_occur) #28294    55
tb16_tax_occur[1:5,1:5]
```

The dataset "Taraspina 16S miTags" contains reads from **45 samples of Tara Oceans** and from **10 samples of Malaspina**. On average, each sample contains 77920 OTUs:

```{r summary_16S, echo=FALSE, message=FALSE, warning=F}
amplicons_per_sample_tb16<-colSums(tb16_tax_occur)
summary(amplicons_per_sample_tb16) #median of amplicons per sample = ~83020)
```

```{r samples_selection_16S, echo=FALSE, message=FALSE, results="hide", warning=F}
#we'll remove all samples with less than 36155 reads.
sort(amplicons_per_sample_tb16[which(colSums(tb16_tax_occur)<36155)])

#remove samples with less than 36155 reads
tb16_tax_occur_min36155 <- tb16_tax_occur[,colSums(tb16_tax_occur) >= 36155]
dim(tb16_tax_occur_min36155) #28294    49

#remove samples with omitted in 18S dataset (so that we can compare the relative abundance of 18S and 16S OTUs considering the same samples)
tb16_tax_occur_min36155<-subset(tb16_tax_occur_min36155, select=-c(TARA_100_SUR_0d2_3,TARA_141_SUR_0d2_3,TARA_93_SUR_0d2_3,TARA_94_SUR_0d2_3,MP0528,MP0778))

dim(tb16_tax_occur_min36155) #28294    43

#let's check if we loose any OTU when excluding the mentioned stations.
tb16_tax_occur_min36155<-tb16_tax_occur_min36155[-(which(rowSums(tb16_tax_occur_min36155)==0)),]
dim(tb16_tax_occur_min36155)
```

```{r starting_dataset_16S, echo=FALSE, results="hide"}
#Table dimensions and content outline:

dim(tb16_tax_occur_min36155)
#tb16_tax_occur_min36155[1:5,1:5]
```

```{r reads_per_sample_overview1_16S, echo=FALSE, results="hide"}
#Minimum number of reads per station:

min(colSums(tb16_tax_occur_min36155)) 
#36155
```

```{r reads_per_sample_overview2_16S, echo=FALSE, results="hide"}
#Maximum number of reads per station:

max(colSums(tb16_tax_occur_min36155)) 
# max: 158933
```

```{r reads_per_sample_overview3_16S, echo=FALSE, results="hide"}
#Identification of stations with higher number of reads:

amplicons_per_sample<-colSums(tb16_tax_occur_min36155)
amplicons_per_sample[which(colSums(tb16_tax_occur_min36155)>158000)]
```
\
Overall reads per sample:

``` {r reads_per_sample_overview4_16S, echo=FALSE}
plot(sort(colSums(tb16_tax_occur_min36155)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 2.2) Normalization

\
In order to keep as many samples as possible, we rarefy at **36155 reads per sample**. By that, we loose 10 samples, and after removing the exluded samples in the 18S dataset (to make them comparable), we end up with a **normalized dataset containing 43 samples and 24691 OTUs**.


``` {r species_richness_rarefaction1_16S, echo=FALSE, results="hide"}
#Let's normalize the original dataset by randomly subsampling 36155 reads in each station:

tb16_tax_occur_min36155_t<-t(tb16_tax_occur_min36155)
tb16_tax_occur_ss36155<-rrarefy(tb16_tax_occur_min36155_t, 36155)
```

```{r species_richness_rarefaction2_16S, echo=FALSE, results="hide"}
dim(tb16_tax_occur_ss36155)
tb16_tax_occur_ss36155[1:5,1:5]
```

``` {r species_richness_rarefaction3_16S, echo=F, results="hide"}
#Its content fits with the expected normalization values (36155 reads per station):

rowSums(tb16_tax_occur_ss36155)
```

```{r species_richness_rarefaction4_16S, echo=F, results="hide"}
#Let's check out how many OTUs don't appear in the new table:

length(which(colSums(tb16_tax_occur_ss36155)==0))
```

```{r species_richness_rarefaction5_16S, echo=F, results="hide"}
#let's remove them from the table and take a look at its final dimensions:

tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155[,-(which(colSums(tb16_tax_occur_ss36155)==0))]
tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155_no_cero[mixedorder(row.names(tb16_tax_occur_ss36155_no_cero)),]
dim(tb16_tax_occur_ss36155_no_cero)

#the final dimensions of the normalized table are 43 19726.
#19726 + 4965 = 24691
```

\
Datasets summary:
```{r dataset_summary_16S, echo=T}
dim(tb16_tax) #original dataset

dim(tb16_tax_occur) #original dataset with occurrence data alone

dim(tb16_tax_occur_min36155) #dataset without samples with less than 36155 OTUs

dim(tb16_tax_occur_ss36155_no_cero) #rarefied dataset
```



## 2.3) General community analysis

### 2.3.1) Richness and evenness (Shannon index)

```{r shannon_index1_16S, echo=FALSE}
tb16_tax_occur_ss36155_div <- diversity(tb16_tax_occur_ss36155_no_cero, index="shannon")
```

Most of the samples take Shannon Index values around 6:

```{r shannon_index2_16S, echo=FALSE}
boxplot(tb16_tax_occur_ss36155_div, pch=19, main="Shannon's index of diversity")
plot(sort(tb16_tax_occur_ss36155_div), pch=19, main="Shannon's index of diversity")
```

### 2.3.2) Richness: OTU number

```{r richness_otu_no1_16S, echo=FALSE}
OTUs_per_sample_16S_tax_occur_ss36155<-specnumber(tb16_tax_occur_ss36155_no_cero)
```

Lowest number of OTUs per sample:

```{r richness_otu_no2_16S, echo=FALSE}
min(OTUs_per_sample_16S_tax_occur_ss36155)
```

Maximum number of OTUs per sample:

```{r richness_otu_no3_16S, echo=FALSE}
max(OTUs_per_sample_16S_tax_occur_ss36155)
```

In most of the samples, we can identify around 4000 OTUs:

```{r richness_otu_no4_16S, echo=F}
plot(sort(OTUs_per_sample_16S_tax_occur_ss36155), pch=19)
boxplot(OTUs_per_sample_16S_tax_occur_ss36155, pch=19)
```

### 2.3.3) Index of evenness

#### 2.3.3.1) Pielou's index

```{r pielou_index_of_evenness1_16S, echo=F, results="hide"}
pielou_evenness_16S_tax_occur_ss36155<-tb16_tax_occur_ss36155_div/log(OTUs_per_sample_16S_tax_occur_ss36155)
```

The Pielou index (constrained between 0 and 1) takes values closer to 1 as the variation of species proportion in a sample increases. Most of the samples get values around 0.85, meaning that the numerical composition of different OTUs in a sample is highly variable - there's no constant presence of dominant species.


```{r pielou_index_of_evenness2_16S, echo=F, results="hide"}
plot(sort(pielou_evenness_16S_tax_occur_ss36155), pch=19)
boxplot(pielou_evenness_16S_tax_occur_ss36155, pch=19)
```


```{r OTUs_overall_abundance_16S, echo=F, results="hide"}
#The OTU_38, with 874 reads, is the most abundant in the overall dataset:

head(sort(colSums(tb16_tax_occur_ss36155_no_cero), decreasing=T), n=10L)
```

Most of the OTUs show very few occurrences, suggesting that we will probably be able to identify a significant ammount of rare otus:

```{r OTUs_overall_abundance2_16S, echo=F, results="hide"}
#previous version:
#plot(log(sort(colSums(tb16_tax_occur_ss36155_no_cero), decreasing=T)), pch=19)

tb16_tax_occur_ss36155_no_cero_rank<-as.data.table(t(tb16_tax_occur_ss36155_no_cero))
tb16_tax_occur_ss36155_no_cero_rank[,V1:=rowSums(tb16_tax_occur_ss36155_no_cero_rank)]
tb16_tax_occur_ss36155_no_cero_rank<-tb16_tax_occur_ss36155_no_cero_rank[order(V1,decreasing = T)]
tb16_tax_occur_ss36155_no_cero_rank[,rank:=1:nrow(tb16_tax_occur_ss36155_no_cero_rank)]

p1<-ggplot(tb16_tax_occur_ss36155_no_cero_rank, aes(x=rank,y=V1))+
  geom_point(aes(x=rank,y=V1))+
  scale_y_log10(name="abundance (log scale)")+
  labs(title="TP 16S rank abundance curve")+
  theme_bw()+
  labs(colour = "") 
p1
```

<!---
#### 1.3.3.2) Sads
library(sads)
?sads
--->

### 2.3.4) Abundance Models
#### 2.3.4.1) Rank-Abundance or Dominance/Diversity Model ("radfit")

The OTUs abundance distribution fits relativelly close to log-normal model. 

```{r radfit_16S, echo=FALSE}
tb16_tax_occur_min36155_radfit<-radfit(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_min36155_radfit)
```

#### 2.3.4.2) Preston's Lognormal Model

According to Preston's lognormal model fit into species frequencies groups, we're missing 83761.1 species:

```{r preston_model1_16S, echo=F}
tb16_tax_occur_ss36155_prestonfit<-prestonfit(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_ss36155_prestonfit, main="TP 18S fit to Preston log-normal model \n(pooled species)")

veiledspec(tb16_tax_occur_ss36155_prestonfit)
```

\
When computing Prestons' lognormal model fit without pooling data into groups, we seem to miss 50576.64 species:

```{r preston_model2_16S, echo=4}
tb16_tax_occur_ss36155_dist_all<-prestondistr(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_ss36155_prestonfit, main="TP 18S fit to Preston log-normal model \n(red: pooled; blue: non-pooled)")
lines(tb16_tax_occur_ss36155_dist_all, line.col="blue3")

veiledspec(tb16_tax_occur_ss36155_dist_all)
```


### 2.3.5) Rarefaction curves of rarefied and non-rarefied datasets

```{r rarefraction_curve_16S, echo=FALSE}
rarec_input<-t(as.matrix(colSums(tb16_tax_occur_ss36155_no_cero)))

tb16_rarecurve_step100<-rarecurve(rarec_input, step = 1000, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="16S amplicons diversity, step=100\n(19,726 OTUs; 1,554,665 reads)\n")

#without rarefying
rarec_allOTUs_input<-t(as.matrix(colSums(t(tb16_tax_occur))))
tb16_rarecurve_allOTUs_step100_8642<-rarecurve(rarec_allOTUs_input, step = 1000, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="16S amplicons diversity non-rarefied, step=1000\n(28,294 OTUs; 4,285,523 reads)\n")
```


### 2.3.6) Beta diversity

#### 2.3.6.1) Dissimilarity matrix using Bray-Curtis index:

The Bray-Curtis dissimilarity, constrained between 0 (minimum distance) and 1 (highest dissimilarity) allows us to quantify the differences between samples according to the composition and relative abundance of their OTUs. In our dataset, most of the samples pairs take dissimilarity values around 0.5, meaning that their composition is substantially different.

```{r beta_div1_16S, echo=FALSE}
#?vegdist
tb16_tax_occur_ss36155_no_cero.bray<-vegdist(tb16_tax_occur_ss36155_no_cero, method="bray")
boxplot(tb16_tax_occur_ss36155_no_cero.bray, main="Bray-Curtis dissimilarity matrix")
```

#### 2.3.6.2) Hierarchical clustering

The stations seem to form clusters according to geographic localization, but there are no evident clusters separated from the general groups, apart from the one containing TARA_84, TARA_82 and TARA_85.

(To be done: assign Longhurst provinces information to each station and check if any of the central clusters is meaningful regarding to the samples' geographical location)

```{r beta_div2_16S, echo=FALSE}
#UPGMA
tb16_tax_occur_ss36155_no_cero.upgma<-hclust(tb16_tax_occur_ss36155_no_cero.bray, "average")
plot(tb16_tax_occur_ss36155_no_cero.upgma, cex=.35, main="Samples Hierarchical Clustering")
```


```{r hclust_coloured_16S, echo=FALSE}
#we'll repeat the UPGMA with the Longhurst codes added in the stations name
longhurst<-read.table("input/TP_longhurst_provinces_merged.txt",header=F,sep="\t")
row.names(longhurst)<-longhurst$V1
tb16_tax_occur_ss36155_no_cero_dend<-merge(tb16_tax_occur_ss36155_no_cero, longhurst, by="row.names", all.x=TRUE)
row.names(tb16_tax_occur_ss36155_no_cero_dend)<-tb16_tax_occur_ss36155_no_cero_dend$V3
tb16_tax_occur_ss36155_no_cero_dend<-tb16_tax_occur_ss36155_no_cero_dend[ , -which(names(tb16_tax_occur_ss36155_no_cero_dend) %in% c("Row.names","V1","V2","V3"))]

tb16_tax_occur_ss36155_no_cero_dend.bray<-vegdist(tb16_tax_occur_ss36155_no_cero_dend, method="bray")
tb16_tax_occur_ss36155_no_cero_dend.upgma<-hclust(tb16_tax_occur_ss36155_no_cero_dend.bray, "average")
plot(tb16_tax_occur_ss36155_no_cero_dend.upgma, cex=.35, main="Samples Hierarchical Clustering")
```

<!--
```{r hclust_coloured2_16S, echo=F, results="hide"}
dend <- as.dendrogram(tb16_tax_occur_ss36155_no_cero_dend.upgma)

###########

# Create a vector giving a color for each province
provinces <- rep("Other", length(rownames(tb16_tax_occur_ss36155_no_cero_dend)))
is_x <- grepl("SATL", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "SATL"
is_x <- grepl("SPSG", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "SPSG"
is_x <- grepl("PEQD", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "PEQD"
is_x <- grepl("NASE", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "NASE"
is_x <- grepl("SPO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "SPO"
is_x <- grepl("NPO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "NPO"
is_x <- grepl("NAO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "NAO"
is_x <- grepl("IO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "IO"
is_x <- grepl("SAO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "SAO"
is_x <- grepl("SO", rownames(tb16_tax_occur_ss36155_no_cero_dend))
provinces[is_x] <- "SO"

provinces <- factor(provinces)
n_provincess <- length(unique(provinces))
cols_4 <- colorspace::rainbow_hcl(n_provincess, c = 70, l  = 50)
col_provinces <- cols_4[provinces]

k234 <- cutree(dend, k = 2:4)

labels_colors(dend) <- col_provinces[order.dendrogram(dend)]
dend <- color_branches(dend, k = 4)

par(mar = c(10,4,2,12))
plot(dend, horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_provinces), dend, rowLabels = c(paste0("k = ", 4:2), "Provinces"), horiz = TRUE)
legend(1,-8, legend=c("Indic", "N. Atlantic", "N. Atlantic Subtropical Gyral Province (East)", "N. Pacific Ocean", "Pacific Equatorial Divergence", "S. Atlantic", "S. Atlantic", "Southern Ocean", "S. Pacific", "S. Pacific Subtropical Gyre"), fill = cols_4, cex=0.8, ncol = 3)

#legend(1, 95, legend=c("Line 1", "Line 2"),
#       col=c("red", "blue"), lty=1:2, cex=0.8)
dev.off()
```
-->


#### 2.3.6.3) Non-metric multidimensional scaling

We can identify an evident division between the stations close to Southern Ocean (TARA 82, 84 and 85) and the rest of the samples. The stress parameter takes a value below 0.2, suggesting that the plot is acceptable. 

```{r monoNMDS_16S, echo=F}
#NMDS
tb16_tax_occur_ss36155_no_cero.nmds<-monoMDS(tb16_tax_occur_ss36155_no_cero.bray)
tb16_tax_occur_ss36155_no_cero.nmds
plot(tb16_tax_occur_ss36155_no_cero.nmds, main="monoMDs method")
```

When implementing a most robut function for computing NMDS plots, the result is quiet the same:

```{r metaNMDS_16S, echo=F, results="hide", warning=F}
tb16_tax_occur_ss36155_no_cero.meta_nmds<-metaMDS(tb16_tax_occur_ss36155_no_cero.bray)
plot(tb16_tax_occur_ss36155_no_cero.meta_nmds, main="metaMDS method")
```

## 2.4) Geographical analysis

```{r load_geo_data_16S, echo=F, results="hide", message=F, warning=F}
#load geographical ubication of stations and sort according to otu_tb16 stations sequence.
MP_geo_16S<-read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/Taraspina_SUR_loc_nodot_stand.name2.txt", sep="\t", header=T)

MP_geo_16S<-data.table(MP_geo_16S)
#MP_geo_16S[,st_name:=(paste("st",MP_geo_16S$station,"_MD",MP_geo_16S$code,sep=""))]
MP_geo_16S<-MP_geo_16S[mixedorder(MP_geo_16S$sample),]
tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155_no_cero[order(row.names(tb16_tax_occur_ss36155_no_cero)),]
#select only stations apearing in tb16_tax_occur_ss36155_no_cero:
MP_geo_16S<-MP_geo_16S[MP_geo_16S$sample %in% row.names(tb16_tax_occur_ss36155_no_cero),]
row.names(MP_geo_16S)<-MP_geo_16S$sample

dim(MP_geo_16S)
MP_geo_16S[1:5,1:3]
tb16_tax_occur_ss36155_no_cero[1:5,1:5]

#we'll use a function from library(fossil) to read lat-long in decimal degrees and translate into distance in km. We'll print only columns containing info about station, latitude and longitude:
MP_geo_16S_reduced<-create.lats(MP_geo_16S, loc="sample", long="long", lat="lat")
head(MP_geo_16S_reduced)

#create a distance matrix (lower triangle) between a list of points.
geo_distances_MP_16S<-earth.dist(MP_geo_16S_reduced, dist = TRUE)
head(geo_distances_MP_16S)
dim(geo_distances_MP_16S)

geo_distances_MP_16S<-as.matrix(geo_distances_MP_16S)
dim(geo_distances_MP_16S)

#geo distances dataset ready to use: "geo_distances_MP_16S"
```

```{r working_datasets1_16S, echo=F, results="hide", warning=F}
#Working datasets:

#1) Community matrix: tb16_tax_occur_ss36155_no_cero
dim(tb16_tax_occur_ss36155_no_cero)
tb16_tax_occur_ss36155_no_cero[1:5, 1:5]

#2) Community Bray-Curtis: tb16_tax_occur_ss36155_no_cero.bray
tb16_tax_occur_ss36155_no_cero.bray<-vegdist(tb16_tax_occur_ss36155_no_cero, method="bray")
tb16_tax_occur_ss36155_no_cero.bray<-as.matrix(tb16_tax_occur_ss36155_no_cero.bray)
dim(tb16_tax_occur_ss36155_no_cero.bray)

#3) Stations distances in km: geo_distances_MP_16S
dim(geo_distances_MP_16S)
```

Communities quickly change their composition across geographical distances:

```{r working_datasets4_16S, echo=F, warning=F}
plot(geo_distances_MP_16S, tb16_tax_occur_ss36155_no_cero.bray, pch=19, cex=0.4, xlab="Geopgraphical distances", ylab="Bray-Curtis dissimilarities")
```

### 2.4.1) Mantel correlograms

Mantel statistic is -significantlly- so low, meaning that the correlation between samples dissimilarity and geographical distances is weak.

```{r mantel_correlogram1_16S, echo=F}
mantel(geo_distances_MP_16S, tb16_tax_occur_ss36155_no_cero.bray)
```

```{r mantel_correlogram2_16S, echo=F, results="hide"}
#Maximum distance between samples:
max(geo_distances_MP_16S)

#Minimum distance between samples:
min(geo_distances_MP_16S)
```

Correlograms:

```{r mantel_correlogram4_16S, echo=T}
MP_16s_ss36155_mantel_correl_by_1000km<-mantel.correlog(tb16_tax_occur_ss36155_no_cero.bray, D.geo=geo_distances_MP_16S, break.pts=seq(0,20000, by=1000))
plot(MP_16s_ss36155_mantel_correl_by_1000km)

MP_16s_ss36155_mantel_correl_by_100km<-mantel.correlog(tb16_tax_occur_ss36155_no_cero.bray, D.geo=geo_distances_MP_16S, break.pts=seq(0,20000, by=100))
plot(MP_16s_ss36155_mantel_correl_by_100km)
```

## 2.5) Abundance vs. occurence

```{r OTUs_mean_relative_abund_16S, echo=F, results="hide"}
tb16_tax_occur_ss36155_no_cero[1:5,1:5]
tb16_tax_occur_ss36155_no_cero_t<-t(tb16_tax_occur_ss36155_no_cero)

colSums(tb16_tax_occur_ss36155_no_cero_t)

#local abundance percentage
tb16_tax_occur_ss36155_no_cero_t.rabund<-tb16_tax_occur_ss36155_no_cero_t/36155

colSums(tb16_tax_occur_ss36155_no_cero_t.rabund)
tb16_tax_occur_ss36155_no_cero_t.rabund[1:5,1:5]

#OTUs mean relative abundance
tb16_tax_occur_ss36155_no_cero_t.rabund_means<-rowMeans(tb16_tax_occur_ss36155_no_cero_t.rabund) 
tb16_tax_occur_ss36155_no_cero_t.rabund_means<-as.data.frame(tb16_tax_occur_ss36155_no_cero_t.rabund_means)

head(tb16_tax_occur_ss36155_no_cero_t.rabund_means)
```

```{r OTUs_occurence_16S, echo=F, results='hide'}
tb16_tax_occur_ss36155_no_cero_t.rabund.occur<-tb16_tax_occur_ss36155_no_cero_t.rabund
tb16_tax_occur_ss36155_no_cero_t.rabund.occur[tb16_tax_occur_ss36155_no_cero_t.rabund.occur>0]<-1
tb16_tax_occur_ss36155_no_cero_t.rabund.occur[1:5,1:5] # presence - absence table

#percentage of occurence in overall stations
tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc<-as.data.frame(100*(rowSums(tb16_tax_occur_ss36155_no_cero_t.rabund.occur)/43))

str(tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc)
```

```{r merge_rabund_peroccur_16S, echo=F, results='hide'}
otu_tb16_ss36155_rabund_percoccur<-merge(tb16_tax_occur_ss36155_no_cero_t.rabund_means,tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc, by="row.names")

colnames(otu_tb16_ss36155_rabund_percoccur)<-c("OTUs","mean_rabund","perc_occur")
otu_tb16_ss36155_rabund_percoccur[1:5,]

row.names(otu_tb16_ss36155_rabund_percoccur)<-otu_tb16_ss36155_rabund_percoccur[,1]
otu_tb16_ss36155_rabund_percoccur<-otu_tb16_ss36155_rabund_percoccur[,-1]
otu_tb16_ss36155_rabund_percoccur[1:5,]
```


OTUs distribution according to their percentage of occurence and relative abundance.\
- red line: OTUs that occur in more than 80% of the samples.\
- blue line: regionally abundant OTUs (> 0.1%).\
- green line: regionally rare OTUs (< 0.001%).


```{r abund_vs_occurence_table_16S, echo=F}
plot(otu_tb16_ss36155_rabund_percoccur$mean_rabund,otu_tb16_ss36155_rabund_percoccur$perc_occur, log="x", pch=19, cex=0.8, xlab="Mean relative abundance", ylab="Percentage of occurence")
abline(h=80, col="red") #occurence higher than 80%
abline(v=0.00001, col="green") #rare OTUs
abline(v=0.001, col="blue") #cosmopolitan OTUs

#Conventional limits:
#Regionally rare     = 0.00001
#Regionally abundant = 0.001
```

Regionally abundant OTUs (relative abundance over 0.1%):

```{r abundant_OTUs_16S, echo=F}
#regionally abundant
tb16_ss36155_abundant<-otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund > 0.001,]

tb16_ss36155_abundant_sorted<-tb16_ss36155_abundant[order(tb16_ss36155_abundant$mean_rabund, tb16_ss36155_abundant$perc_occur, decreasing = T), c(1,2)]

tb16_ss36155_abundant_sorted_prov<-cbind(otu_names=row.names(tb16_ss36155_abundant_sorted),tb16_ss36155_abundant_sorted)
tb16_class_prov<-cbind(otu_names=row.names(tb16_class),tb16_class)
#tb16_class_prov<-tb16_class_prov[,-2]
tb16_ss36155_abundant_sorted_prov<-merge(tb16_ss36155_abundant_sorted_prov,tb16_class_prov, by="otu_names", all.x=TRUE)

tb16_ss36155_abundant_sorted_prov<-tb16_ss36155_abundant_sorted_prov[order( tb16_ss36155_abundant_sorted_prov$mean_rabund, tb16_ss36155_abundant_sorted_prov$perc_occur, decreasing = T), c(1,2,3,6,4)]
tb16_ss36155_abundant_sorted_prov
#dim(tb16_ss36155_abundant_sorted_prov)
```

Number and roportion of regionally abundant OTUs (%):

```{r abundant_OTUs2_16S, echo=F}
nrow(tb16_ss36155_abundant_sorted_prov)

(nrow(tb16_ss36155_abundant_sorted_prov)/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```

Cosmopolitan OTUs (relative abundance over 0.1% and occurence in more than 80% of samples):

```{r select_cosmopolitan_16S, echo=F}
otu_tb16_ss36155_rabund_cosm<-otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund > 0.001,]
otu_tb16_ss36155_rabund_poccur_cosm<-otu_tb16_ss36155_rabund_cosm[otu_tb16_ss36155_rabund_cosm$perc_occur > 80,]
otu_tb16_ss36155_cosmop_sorted<-otu_tb16_ss36155_rabund_poccur_cosm[order(otu_tb16_ss36155_rabund_poccur_cosm$perc_occur, otu_tb16_ss36155_rabund_poccur_cosm$mean_rabund, decreasing = T), c(1,2)]


otu_tb16_ss36155_cosmop_sorted_prov<-cbind(otu_names=row.names(otu_tb16_ss36155_cosmop_sorted),otu_tb16_ss36155_cosmop_sorted)
otu_tb16_ss36155_cosmop_sorted_prov<-merge(otu_tb16_ss36155_cosmop_sorted_prov,tb16_class_prov, by="otu_names", all.x=TRUE)

otu_tb16_ss36155_cosmop_sorted_prov<-otu_tb16_ss36155_cosmop_sorted_prov[order(otu_tb16_ss36155_cosmop_sorted_prov$perc_occur, otu_tb16_ss36155_cosmop_sorted_prov$mean_rabund, decreasing = T), c(1,2,3,6,4)]
otu_tb16_ss36155_cosmop_sorted_prov
#dim(otu_tb16_ss36155_cosmop_sorted_prov)
```

Number and proportion (%) of cosmopolitan OTUs:

```{r percentage_cosmopolitan_16S, echo=F}
nrow(otu_tb16_ss36155_cosmop_sorted_prov)

(nrow(otu_tb16_ss36155_cosmop_sorted_prov)/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```

Number and proportion (%) of rare OTUs:

```{r rare_OTUs_16S, echo=F}
nrow(otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund < 0.00001 & otu_tb16_ss36155_rabund_percoccur$mean_rabund >0,])

(nrow(otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund < 0.00001 & otu_tb16_ss36155_rabund_percoccur$mean_rabund >0,])/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```




## 2.6) Taxonomic composition analysis

### 2.6.1) Normalized data

```{r merge_tables_16S, echo=FALSE, results="hide", warning=FALSE}
#Let's add the taxonomic classification by merging "tb16_tax_occur_ss36155_no_cero" with "tb16_tax":

tb16_tax_occur_ss36155_no_cero_t<-t(tb16_tax_occur_ss36155_no_cero)
tb16_ss36155_tax<-merge(tb16_tax_occur_ss36155_no_cero_t,tb16_class, by="row.names", all.x=T)

dim(tb16_ss36155_tax)
tb16_ss36155_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb16_ss36155_tax)=tb16_ss36155_tax$Row.names
#add OTU_no as rowname
rownames.tb16_ss36155_tax<-tb16_ss36155_tax[,1]
tb16_ss36155_tax<-tb16_ss36155_tax[,-1]

dim(tb16_ss36155_tax)
tb16_ss36155_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb16_ss36155_tax["OTU_no"] <- NA
tb16_ss36155_tax$OTU_no <- sapply(strsplit(rownames(tb16_ss36155_tax),split= "\\_"),'[',2)
tb16_ss36155_tax$OTU_no <- as.numeric(as.character(tb16_ss36155_tax$OTU_no))
tb16_ss36155_tax_sorted<-tb16_ss36155_tax[order(tb16_ss36155_tax$OTU_no, decreasing = FALSE), ]
```

No. of OTUs and reads of the rearefied dataset:

```{r dim_merge1_16S, echo=F}
nrow(tb16_ss36155_tax_sorted)
sum(colSums(tb16_ss36155_tax_sorted[,1:43]))

#dim(tb16_ss36155_tax_sorted)
#tb16_ss36155_tax_sorted[1:5,1:5]
```

No. of OTUs and reads of phototrophic groups:

```{r dim_merge2_16S, echo=F}
#selection of phototrophs:
tb16_phototrophs <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A != "other_bacteria"),]

nrow(tb16_phototrophs)
sum(colSums(tb16_phototrophs[,1:43]))

#dim(tb16_phototrophs)
#tb16_phototrophs[1:5,1:5]
```


No. of OTUs and reads of non-phototrophic groups:

```{r dim_merge3_16S, echo=F}
#selection of non.phototrophs:
tb16_non.phototrophs <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_bacteria"),]

nrow(tb16_non.phototrophs)
sum(colSums(tb16_non.phototrophs[,1:43]))

#dim(tb16_non.phototrophs)
#tb16_non.phototrophs[1:5,1:5]
```

\
**PHOTOTROPHS + HETEROTROPHS**

**Absolute values**


```{r count_samples_per_group_Heterotrophs_16S, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

Cyanobacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:43]
Cyanobacteria_tb_occur_len<-length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cyanobacteria",samples_per_class=Cyanobacteria_tb_occur_len))

other_cyanobacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_cyanobacteria"),]
other_cyanobacteria_tb_occur <- other_cyanobacteria_tb[,1:43]
other_cyanobacteria_tb_occur_len<-length(other_cyanobacteria_tb_occur[,colSums(other_cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_cyanobacteria",samples_per_class=other_cyanobacteria_tb_occur_len))

other_bacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_bacteria"),]
other_bacteria_tb_occur <- other_bacteria_tb[,1:43]
other_bacteria_tb_occur_len<-length(other_bacteria_tb_occur[,colSums(other_bacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_bacteria",samples_per_class=other_bacteria_tb_occur_len))

other_plastids_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_plastids"),]
other_plastids_tb_occur <- other_plastids_tb[,1:43]
other_plastids_tb_occur_len<-length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_plastids",samples_per_class=other_plastids_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

```{r aggregate_Heterotrophs_16S, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb16_ss36155_tax_sorted[1:43]), list(tb16_ss36155_tax_sorted$class_A), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb16_ss36155_tax_sorted[1:43]), list(tb16_ss36155_tax_sorted$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_Heterotrophs_16S, echo=FALSE, warning=F}
tb16S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
colnames(tb16S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb16S_OTUs_reads_samples[1:5,1:2]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[order(tb16S_OTUs_reads_samples$reads_per_class, tb16S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_OTUs_reads_samples<-merge(tb16S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]
tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
tb16S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_16S_Heterotrophs, echo=FALSE, warning=F}

tb16S_OTUs_reads_samples_rel_abund <- tb16S_OTUs_reads_samples

tb16S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_OTUs_reads_samples)[1]
tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_OTUs_reads_samples)[2]
tb16S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb16S_OTUs_reads_samples_rel_abund)
tb16S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_16S_v2_Heterotrophs, echo=FALSE, warnings=F, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-16S OTUs (%)", y="TP-miTags-16S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_16S_Heterotrophs, echo=FALSE, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-16S samples (%)", y="TP-miTags-16S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```



\
**PHOTOTROPHS**

```{r count_samples_per_group_16S, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

other_plastids_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "other_plastids"),]
other_plastids_tb_occur <- other_plastids_tb[,1:43]
other_plastids_tb_occur_len<-length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_plastids",samples_per_class=other_plastids_tb_occur_len))

Cyanobacteria_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:43]
Cyanobacteria_tb_occur_len<-length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cyanobacteria",samples_per_class=Cyanobacteria_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```


**Absolute values**

```{r aggregate_16S, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb16_phototrophs[1:43]), list(tb16_phototrophs$class_A), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb16_phototrophs[1:43]), list(tb16_phototrophs$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_16S, echo=FALSE, warning=F}
tb16S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
colnames(tb16S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb16S_OTUs_reads_samples[1:5,1:2]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[order(tb16S_OTUs_reads_samples$reads_per_class, tb16S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_OTUs_reads_samples<-merge(tb16S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]
tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
tb16S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_16S, echo=FALSE, warning=F}

tb16S_OTUs_reads_samples_rel_abund <- tb16S_OTUs_reads_samples

tb16S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_OTUs_reads_samples)[1]
tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_OTUs_reads_samples)[2]
tb16S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb16S_OTUs_reads_samples_rel_abund)
tb16S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_16S_v2, echo=FALSE, warnings=F, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-16S OTUs (%)", y="TP-miTags-16S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_16S, echo=FALSE, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-16S samples (%)", y="TP-miTags-16S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


```{r cyanobacteria_histogram, echo=F, results="hide", warning=F}
cyano_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cyanobacteria"),]

class_summary_reads_per_class_cyano<-aggregate(rowSums(cyano_tb[1:43]), list(cyano_tb$class_B), sum)
# count the different groups
class_summary_otus_per_class_cyano<-aggregate(rowSums(cyano_tb[1:43]), list(cyano_tb$class_B), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano[order(-x),]
detach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order

#fix column names
row.names(class_summary_reads_per_class_cyano_order)<-class_summary_reads_per_class_cyano_order[,1]
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano_order[c(-1)]
colnames(class_summary_reads_per_class_cyano_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano[order(-x),]
detach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order

row.names(class_summary_otus_per_class_cyano_order)<-class_summary_otus_per_class_cyano_order[,1]
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano_order[c(-1)]
colnames(class_summary_otus_per_class_cyano_order)<-c("OTUs_per_class")

# create_table_merging_#OTUs_#reads
cyano_OTUs_reads <- merge(class_summary_reads_per_class_cyano_order, class_summary_otus_per_class_cyano_order, by="row.names")

row.names(cyano_OTUs_reads)<-cyano_OTUs_reads[,1]
cyano_OTUs_reads<-cyano_OTUs_reads[,-1]
colnames(cyano_OTUs_reads)<-c("reads_per_class","OTUs_per_class")
cyano_OTUs_reads[1:3,1:2]

cyano_OTUs_reads<-cyano_OTUs_reads[order(cyano_OTUs_reads$reads_per_class, cyano_OTUs_reads$OTUs_per_class, decreasing = T), c(1,2)]
cyano_OTUs_reads

#compute relative values
cyano_OTUs_reads_rel_abund <- cyano_OTUs_reads

cyano_OTUs_reads_rel_abund$reads_per_class<-(cyano_OTUs_reads_rel_abund$reads_per_class*100)/colSums(cyano_OTUs_reads)[1]
cyano_OTUs_reads_rel_abund$OTUs_per_class<-(cyano_OTUs_reads_rel_abund$OTUs_per_class*100)/colSums(cyano_OTUs_reads)[2]

colSums(cyano_OTUs_reads_rel_abund)
rownames(cyano_OTUs_reads_rel_abund) = c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")
cyano_OTUs_reads_rel_abund

cyano_OTUs_reads_rel_abund["cyano_group"]<-NA
cyano_OTUs_reads_rel_abund$cyano_group<-c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")

cyano_OTUs_reads_rel_abund2<-read.table("input/cyano_histograms_data.txt", head=TRUE)
cyano_OTUs_reads_rel_abund2

ggplot(cyano_OTUs_reads_rel_abund2, aes(x=group, y=value, fill=data)) + 
  geom_bar(position="dodge", stat="identity") +
  scale_y_continuous(limits=c(-1,100), breaks=c(25,50,75,100)) +
  scale_fill_manual(values=c("#000000", "cadetblue3")) +
  labs(x="", y="% \n") + 
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16)) + 
  theme(legend.title = element_blank(), legend.position=c(0.85,0.85), legend.text=element_text(size=14)) +
  theme(axis.title.y = element_text(size = rel(1.5), angle = 0, vjust=0.97))
```



### 2.6.2) Non-rarefied data


```{r merge_tables_non.raref_16S, echo=FALSE, results="hide", warning=FALSE}
#Let's add the taxonomic class_Aication by merging "tb16_tax_occur_min36155_no_cero" with "tb16_tax":

tb16_tax_occur_min36155_t<-tb16_tax_occur_min36155
tb16_min36155_tax<-merge(tb16_tax_occur_min36155_t,tb16_class, by="row.names")


dim(tb16_min36155_tax)
tb16_min36155_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb16_min36155_tax)=tb16_min36155_tax$Row.names
#add OTU_no as rowname
rownames.tb16_min36155_tax<-tb16_min36155_tax[,1]
tb16_min36155_tax<-tb16_min36155_tax[,-1]

dim(tb16_min36155_tax)
tb16_min36155_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb16_min36155_tax["OTU_no"] <- NA
tb16_min36155_tax$OTU_no <- sapply(strsplit(rownames(tb16_min36155_tax),split= "\\_"),'[',2)
tb16_min36155_tax$OTU_no <- as.numeric(as.character(tb16_min36155_tax$OTU_no))
tb16_min36155_tax_sorted<-tb16_min36155_tax[order(tb16_min36155_tax$OTU_no, decreasing = FALSE), ]
```

No. of OTUs and reads of the rearefied dataset:

```{r dim_merge1_non.raref_16S, echo=F}
nrow(tb16_min36155_tax_sorted)
sum(colSums(tb16_min36155_tax_sorted[,1:43]))

#dim(tb16_min36155_tax_sorted)
#tb16_min36155_tax_sorted[1:5,1:5]
```

No. of OTUs and reads of phototrophic groups:

```{r dim_merge2_non.raref_16S, echo=F}
#selection of phototrophs:
tb16_phototrophs <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A != "other_bacteria"),]

nrow(tb16_phototrophs)
sum(colSums(tb16_phototrophs[,1:43]))

#dim(tb16_phototrophs)
#tb16_phototrophs[1:5,1:5]
```


No. of OTUs and reads of non-phototrophic groups:

```{r dim_merge3_non.raref_16S, echo=F}
#selection of non.phototrophs:
tb16_non.phototrophs <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "other_bacteria"),]

nrow(tb16_non.phototrophs)
sum(colSums(tb16_non.phototrophs[,1:43]))

#dim(tb16_non.phototrophs)
#tb16_non.phototrophs[1:5,1:5]
```


\
**PHOTOTROPHS + HETEROTROPHS**

**Absolute values**


```{r count_samples_per_group_Heterotrophs_non.raref_16S, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb16_min36155_tax_sorted[which(tb16_min36155_tax_sorted$class_A == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

Cyanobacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:43]
Cyanobacteria_tb_occur_len<-length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cyanobacteria",samples_per_class=Cyanobacteria_tb_occur_len))

other_cyanobacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_cyanobacteria"),]
other_cyanobacteria_tb_occur <- other_cyanobacteria_tb[,1:43]
other_cyanobacteria_tb_occur_len<-length(other_cyanobacteria_tb_occur[,colSums(other_cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_cyanobacteria",samples_per_class=other_cyanobacteria_tb_occur_len))

other_bacteria_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_bacteria"),]
other_bacteria_tb_occur <- other_bacteria_tb[,1:43]
other_bacteria_tb_occur_len<-length(other_bacteria_tb_occur[,colSums(other_bacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_bacteria",samples_per_class=other_bacteria_tb_occur_len))

other_plastids_tb <- tb16_ss36155_tax_sorted[which(tb16_ss36155_tax_sorted$class_A == "other_plastids"),]
other_plastids_tb_occur <- other_plastids_tb[,1:43]
other_plastids_tb_occur_len<-length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_plastids",samples_per_class=other_plastids_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```

```{r aggregate_Heterotrophs_non.raref_16S, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb16_min36155_tax_sorted[1:43]), list(tb16_min36155_tax_sorted$class_A), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb16_min36155_tax_sorted[1:43]), list(tb16_min36155_tax_sorted$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_Heterotrophs_non.raref_16S, echo=FALSE, warning=F}
tb16S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
colnames(tb16S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb16S_OTUs_reads_samples[1:5,1:2]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[order(tb16S_OTUs_reads_samples$reads_per_class, tb16S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_OTUs_reads_samples<-merge(tb16S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]
tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
tb16S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S_Heterotrophs_non.raref_16S, echo=FALSE, warning=F}

tb16S_OTUs_reads_samples_rel_abund <- tb16S_OTUs_reads_samples

tb16S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_OTUs_reads_samples)[1]
tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_OTUs_reads_samples)[2]
tb16S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb16S_OTUs_reads_samples_rel_abund)
tb16S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2_Heterotrophs_non.raref_16S, echo=FALSE, warnings=F, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-16S OTUs (%)", y="TP-miTags-16S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S_Heterotrophs_non.raref_16S, echo=FALSE, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-16S samples (%)", y="TP-miTags-16S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
**PHOTOTROPHS**

```{r count_samples_per_group_non.raref_16S, echo=F, warning=FALSE, results="hide"}
occurrence_counts_phototrophs<-data.table()

#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Dinophyceae"),]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur_len<-length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dinophyceae",samples_per_class=Dinophyceae_tb_occur_len))

Prasinophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
Prasinophyceae_tb_occur_len<-length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_Prasinophyceae",samples_per_class=Prasinophyceae_tb_occur_len))

Chrysophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
Chrysophyceae_tb_occur_len<-length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chrysophyceae",samples_per_class=Chrysophyceae_tb_occur_len))

Pelagophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
Pelagophyceae_tb_occur_len<-length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pelagophyceae",samples_per_class=Pelagophyceae_tb_occur_len))

Dictyochophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
Dictyochophyceae_tb_occur_len<-length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Dictyochophyceae",samples_per_class=Dictyochophyceae_tb_occur_len))

Cryptophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cryptophyceae"),]
Cryptophyceae_tb_occur <- Cryptophyceae_tb[,1:43]
Cryptophyceae_tb_occur_len<-length(Cryptophyceae_tb_occur[,colSums(Cryptophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cryptophyceae",samples_per_class=Cryptophyceae_tb_occur_len))

Bacillariophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Bacillariophyceae"),]
Bacillariophyceae_tb_occur <- Bacillariophyceae_tb[,1:43]
Bacillariophyceae_tb_occur_len<-length(Bacillariophyceae_tb_occur[,colSums(Bacillariophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bacillariophyceae",samples_per_class=Bacillariophyceae_tb_occur_len))

Chlorarachniophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorarachniophyceae"),]
Chlorarachniophyceae_tb_occur <- Chlorarachniophyceae_tb[,1:43]
Chlorarachniophyceae_tb_occur_len<-length(Chlorarachniophyceae_tb_occur[,colSums(Chlorarachniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorarachniophyceae",samples_per_class=Chlorarachniophyceae_tb_occur_len))

Bolidophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
Bolidophyceae_tb_occur_len<-length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Bolidophyceae",samples_per_class=Bolidophyceae_tb_occur_len))

Pinguiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pinguiophyceae"),]
Pinguiophyceae_tb_occur <- Pinguiophyceae_tb[,1:43]
Pinguiophyceae_tb_occur_len<-length(Pinguiophyceae_tb_occur[,colSums(Pinguiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pinguiophyceae",samples_per_class=Pinguiophyceae_tb_occur_len))

Prymnesiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
Prymnesiophyceae_tb_occur_len<-length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prymnesiophyceae",samples_per_class=Prymnesiophyceae_tb_occur_len))

Mamiellophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
Mamiellophyceae_tb_occur_len<-length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Mamiellophyceae",samples_per_class=Mamiellophyceae_tb_occur_len))

Eustigmatophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Eustigmatophyceae"),]
Eustigmatophyceae_tb_occur <- Eustigmatophyceae_tb[,1:43]
Eustigmatophyceae_tb_occur_len<-length(Eustigmatophyceae_tb_occur[,colSums(Eustigmatophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Eustigmatophyceae",samples_per_class=Eustigmatophyceae_tb_occur_len))

Chlorophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
Chlorophyceae_tb_occur_len<-length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorophyceae",samples_per_class=Chlorophyceae_tb_occur_len))

Ulvophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
Ulvophyceae_tb_occur_len<-length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Ulvophyceae",samples_per_class=Ulvophyceae_tb_occur_len))

Raphydophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
Raphydophyceae_tb_occur_len<-length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Raphydophyceae",samples_per_class=Raphydophyceae_tb_occur_len))

Trebouxiophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
Trebouxiophyceae_tb_occur_len<-length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Trebouxiophyceae",samples_per_class=Trebouxiophyceae_tb_occur_len))

Phaeophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
Phaeophyceae_tb_occur_len<-length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeophyceae",samples_per_class=Phaeophyceae_tb_occur_len))

Phaeothamniophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
Phaeothamniophyceae_tb_occur_len<-length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Phaeothamniophyceae",samples_per_class=Phaeothamniophyceae_tb_occur_len))

Xanthophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
Xanthophyceae_tb_occur_len<-length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Xanthophyceae",samples_per_class=Xanthophyceae_tb_occur_len))

Chlorodendrophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
Chlorodendrophyceae_tb_occur_len<-length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Chlorodendrophyceae",samples_per_class=Chlorodendrophyceae_tb_occur_len))

IncertaeSedis_Archaeplastida_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
IncertaeSedis_Archaeplastida_tb_occur_len<-length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="IncertaeSedis_Archaeplastida",samples_per_class=IncertaeSedis_Archaeplastida_tb_occur_len))

Nephroselmidophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
Nephroselmidophyceae_tb_occur_len<-length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Nephroselmidophyceae",samples_per_class=Nephroselmidophyceae_tb_occur_len))

Pavlovophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
Pavlovophyceae_tb_occur_len<-length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pavlovophyceae",samples_per_class=Pavlovophyceae_tb_occur_len))

Rhodophyceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
Rhodophyceae_tb_occur_len<-length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rhodophyceae",samples_per_class=Rhodophyceae_tb_occur_len))

Rappemonads_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
Rappemonads_tb_occur_len<-length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Rappemonads",samples_per_class=Rappemonads_tb_occur_len))

MOCH_1_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
MOCH_1_tb_occur_len<-length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_1",samples_per_class=MOCH_1_tb_occur_len))

MOCH_2_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
MOCH_2_tb_occur_len<-length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_2",samples_per_class=MOCH_2_tb_occur_len))

MOCH_5_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "MOCH_5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
MOCH_5_tb_occur_len<-length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="MOCH_5",samples_per_class=MOCH_5_tb_occur_len))

Prasinophyceae_clade_VII_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
Prasinophyceae_clade_VII_tb_occur_len<-length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-VII",samples_per_class=Prasinophyceae_clade_VII_tb_occur_len))

Prasinophyceae_clade_IX_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
Prasinophyceae_clade_IX_tb_occur_len<-length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Prasinophyceae_clade-IX",samples_per_class=Prasinophyceae_clade_IX_tb_occur_len))

Pyramimonadaceae_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
Pyramimonadaceae_tb_occur_len<-length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Pyramimonadaceae",samples_per_class=Pyramimonadaceae_tb_occur_len))

other_plastids_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "other_plastids"),]
other_plastids_tb_occur <- other_plastids_tb[,1:43]
other_plastids_tb_occur_len<-length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="other_plastids",samples_per_class=other_plastids_tb_occur_len))

Cyanobacteria_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:43]
Cyanobacteria_tb_occur_len<-length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
occurrence_counts_phototrophs<-rbind(occurrence_counts_phototrophs,data.table(group="Cyanobacteria",samples_per_class=Cyanobacteria_tb_occur_len))

occurrence_counts_phototrophs
#row.names(occurrence_counts_phototrophs)<-occurrence_counts_phototrophs$group
occurrence_counts_phototrophs<-as.data.frame(occurrence_counts_phototrophs)
```


**Absolute values**

```{r aggregate_non.raref_16S, echo=FALSE, results="hide", warning=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb16_phototrophs[1:43]), list(tb16_phototrophs$class_A), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb16_phototrophs[1:43]), list(tb16_phototrophs$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_non.raref_16S, echo=FALSE, warning=F}
tb16S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
colnames(tb16S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
#tb16S_OTUs_reads_samples[1:5,1:2]

tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[order(tb16S_OTUs_reads_samples$reads_per_class, tb16S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_OTUs_reads_samples<-merge(tb16S_OTUs_reads_samples, occurrence_counts_phototrophs, by.x="row.names", by.y="group", all.x=TRUE)

row.names(tb16S_OTUs_reads_samples)<-tb16S_OTUs_reads_samples[,1]
tb16S_OTUs_reads_samples<-tb16S_OTUs_reads_samples[,-1]
tb16S_OTUs_reads_samples
```


\
**Relative values**

```{r OTUs_vs_reads_relative_values_18S_non.raref_16S, echo=FALSE, warning=F}

tb16S_OTUs_reads_samples_rel_abund <- tb16S_OTUs_reads_samples

tb16S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_OTUs_reads_samples)[1]
tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_OTUs_reads_samples)[2]
tb16S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb16S_OTUs_reads_samples_rel_abund)
tb16S_OTUs_reads_samples_rel_abund
```

\
\

Reads per class vs. OTUs per class:

```{r OTUs_vs_reads_18S_v2_non.raref_16S, echo=FALSE, warnings=F, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[1] Reads per class vs. OTUs per class", x="\nTP-miTags-16S OTUs (%)", y="TP-miTags-16S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\

Reads per class vs. samples in which they occurr:

```{r reads_vs_samples_18S_non.raref_16S, echo=FALSE, warning=F}
ggplot(tb16S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb16S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[2] Reads per class vs. samples in which they occur", x="\nTP-miTags-16S samples (%)", y="TP-miTags-16S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

\
\
\
\

```{r cyanobacteria_histogram_non.raref_16S, echo=F, results="hide", warning=F}
cyano_tb <- tb16_phototrophs[which(tb16_phototrophs$class_A == "Cyanobacteria"),]

class_summary_reads_per_class_cyano<-aggregate(rowSums(cyano_tb[1:43]), list(cyano_tb$class_B), sum)
# count the different groups
class_summary_otus_per_class_cyano<-aggregate(rowSums(cyano_tb[1:43]), list(cyano_tb$class_B), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano[order(-x),]
detach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order

#fix column names
row.names(class_summary_reads_per_class_cyano_order)<-class_summary_reads_per_class_cyano_order[,1]
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano_order[c(-1)]
colnames(class_summary_reads_per_class_cyano_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano[order(-x),]
detach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order

row.names(class_summary_otus_per_class_cyano_order)<-class_summary_otus_per_class_cyano_order[,1]
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano_order[c(-1)]
colnames(class_summary_otus_per_class_cyano_order)<-c("OTUs_per_class")

# create_table_merging_#OTUs_#reads
cyano_OTUs_reads <- merge(class_summary_reads_per_class_cyano_order, class_summary_otus_per_class_cyano_order, by="row.names")

row.names(cyano_OTUs_reads)<-cyano_OTUs_reads[,1]
cyano_OTUs_reads<-cyano_OTUs_reads[,-1]
colnames(cyano_OTUs_reads)<-c("reads_per_class","OTUs_per_class")
cyano_OTUs_reads[1:3,1:2]

cyano_OTUs_reads<-cyano_OTUs_reads[order(cyano_OTUs_reads$reads_per_class, cyano_OTUs_reads$OTUs_per_class, decreasing = T), c(1,2)]
```

\
Absolute values of cyanobacteria groups richness and abundance:
```{r cyanos_abs_values_non.raref, echo=F}
cyano_OTUs_reads
```

```{r cyanos_rel_values_non.raref, echo=F, results="hide"}
#compute relative values
cyano_OTUs_reads_rel_abund <- cyano_OTUs_reads

cyano_OTUs_reads_rel_abund$reads_per_class<-(cyano_OTUs_reads_rel_abund$reads_per_class*100)/colSums(cyano_OTUs_reads)[1]
cyano_OTUs_reads_rel_abund$OTUs_per_class<-(cyano_OTUs_reads_rel_abund$OTUs_per_class*100)/colSums(cyano_OTUs_reads)[2]

colSums(cyano_OTUs_reads_rel_abund)
rownames(cyano_OTUs_reads_rel_abund) = c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")
cyano_OTUs_reads_rel_abund

cyano_OTUs_reads_rel_abund["cyano_group"]<-NA
cyano_OTUs_reads_rel_abund$cyano_group<-c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")

cyano_OTUs_reads_rel_abund2<-read.table("input/cyano_histograms_data_non.raref.txt", head=TRUE)
```


Relative values of Cyanobacteria groups richness and abundance:

```{r print_rel_values_non.raref, echo=F}
cyano_OTUs_reads_rel_abund[,1:2]

ggplot(cyano_OTUs_reads_rel_abund2, aes(x=group, y=value, fill=data)) + 
  geom_bar(position="dodge", stat="identity") +
  scale_y_continuous(limits=c(-1,100), breaks=c(25,50,75,100)) +
  scale_fill_manual(values=c("#000000", "cadetblue3")) +
  labs(x="", y="% \n") + 
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16)) + 
  theme(legend.title = element_blank(), legend.position=c(0.85,0.85), legend.text=element_text(size=14)) +
  theme(axis.title.y = element_text(size = rel(1.5), angle = 0, vjust=0.97))
```



