---
title: "TFM2_Taraspina_miTags_analysis"
author: "lrubinat"
date: "5/23/2017"
output:
  html_document:
    theme: united
    toc: yes
  pdf_document:
    highlight: zenburn
    toc: yes
---


<!--- INITIALIZATION
```{r, echo=FALSE}
#error hook to kill knitr in case of errors
library(knitr)
knit_hooks$set(error = function(x, options) stop(x))
opts_chunk$set(cache=TRUE, autodep=TRUE)
```
--->

```{r libraries, echo=T, message=F}
library(data.table)
library(dplyr)
library(DT)
library(ggplot2)
library(gtools)
library(vegan)
library(iNEXT)
library(fossil)
library(ggrepel)
```



# 1) 18S amplicons

## 1.1) Data overview

Let's read the dataset and remove the samples containing less than 1154 reads:

``` {r load_data, echo=FALSE, message=FALSE}
setwd("~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/")

#read data 
tb18_tax <- read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/otu_table99_SUR_miTags_18S_classif.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(tb18_tax) # 8642   58
tb18_tax[1:5,1:5]

row.names(tb18_tax)<-tb18_tax[,1]
tb18_tax<-tb18_tax[,-1]
tb18_tax[is.na(tb18_tax)]<-"NA"

dim(tb18_tax) #8642   57
tb18_tax[1:5,1:5]

#table with taxonomic classification alone:
tb18_class <- tb18_tax[,56:57]
dim(tb18_class) #8642    2
tb18_class[1:5,1:2]

#table with occurence data alone
tb18_tax_occur <- tb18_tax[,1:55]
dim(tb18_tax_occur) #8642   55
tb18_tax_occur[1:5,1:5]

amplicons_per_sample_tb18<-colSums(tb18_tax_occur)
summary(amplicons_per_sample_tb18) #median of amplicons per sample = ~3086)
#we'll remove all samples with less than 1154 reads.
amplicons_per_sample_tb18[which(colSums(tb18_tax_occur)<1154)]

#remove samples with less than 1154 reads
tb18_tax_occur_min1154 <- tb18_tax_occur[,colSums(tb18_tax_occur) >= 1154]
dim(tb18_tax_occur_min1154) #8642   45

#remove samples with omitted in 16S dataset (so that we can compare the relative abundance of 16S and 18S OTUs considering the same samples)
tb18_tax_occur_min1154<-subset(tb18_tax_occur_min1154, select=-c(TARA_70_SUR_0d2_3,TARA_67_SUR_0d2_3))


dim(tb18_tax_occur_min1154) #8642   43

#let's check if we loose any OTU when excluding the mentioned stations.
tb18_tax_occur_min1154_no_cero<-tb18_tax_occur_min1154[,-(which(colSums(tb18_tax_occur_min1154)==0))]
dim(tb18_tax_occur_min1154_no_cero)
```

Table dimensions and content outline:

```{r starting_dataset, echo=FALSE}
dim(tb18_tax_occur_min1154)
tb18_tax_occur_min1154[1:5,1:5]
```

Minimum number of reads per station:

```{r reads_per_sample_overview1, echo=1}
min(colSums(tb18_tax_occur_min1154)) 
#1154
```

Maximum number of reads per station:

```{r reads_per_sample_overview2, echo=1}
max(colSums(tb18_tax_occur_min1154)) 
# max: 23781
```

Identification of stations with higher number of reads:

```{r reads_per_sample_overview3, echo=TRUE}
amplicons_per_sample<-colSums(tb18_tax_occur_min1154)
amplicons_per_sample[which(colSums(tb18_tax_occur_min1154)>20000)]
```

Overall reads per sample:

``` {r reads_per_sample_overview4, echo=FALSE}
plot(sort(colSums(tb18_tax_occur_min1154)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 1.2) Normalization

Let's normalize the original dataset by randomly subsampling 1154 reads in each station:

``` {r species_richness_rarefaction1, echo=TRUE}
tb18_tax_occur_min1154_t<-t(tb18_tax_occur_min1154)
tb18_tax_occur_ss1154<-rrarefy(tb18_tax_occur_min1154_t, 1154)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2, echo=FALSE}
dim(tb18_tax_occur_ss1154)
tb18_tax_occur_ss1154[1:5,1:5]
```

Its content fits with the expected normalization values (1154 reads per station):

``` {r species_richness_rarefaction3, echo=TRUE}
rowSums(tb18_tax_occur_ss1154)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4, echo=1:5}
length(which(colSums(tb18_tax_occur_ss1154)==0)) #2923
```

There are 2923 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5, echo=1:3}
tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154[,-(which(colSums(tb18_tax_occur_ss1154)==0))]
tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154_no_cero[mixedorder(row.names(tb18_tax_occur_ss1154_no_cero)),]
dim(tb18_tax_occur_ss1154_no_cero) #43 5719

#the final dimensions of the normalized table are 43 5719.
#5719 + 2923 = 8642
```

Datasets summary:
```{r dataset_summary}
dim(tb18_tax) #8642   57
dim(tb18_tax_occur) #8642   55
dim(tb18_tax_occur_ss1154_no_cero) #43 5719
```



## 1.3) General community analysis

### 1.3.1) Richness and evenness (Shannon index)

```{r shannon_index1, echo=FALSE}
tb18_tax_occur_ss1154_div <- diversity(tb18_tax_occur_ss1154_no_cero, index="shannon")
```

Most of the samples take Shannon Index values around 6:

```{r shannon_index2, echo=FALSE}
boxplot(tb18_tax_occur_ss1154_div, pch=19, main="Shannon's index of diversity")
plot(sort(tb18_tax_occur_ss1154_div), pch=19, main="Shannon's index of diversity")
```

### 1.3.2) Richness: OTU number

```{r richness_otu_no1, echo=FALSE}
OTUs_per_sample_18S_tax_occur_ss1154<-specnumber(tb18_tax_occur_ss1154_no_cero)
```

Lowest number of OTUs per sample:

```{r richness_otu_no2, echo=FALSE}
min(OTUs_per_sample_18S_tax_occur_ss1154)
```

Maximum number of OTUs per sample:

```{r richness_otu_no3, echo=FALSE}
max(OTUs_per_sample_18S_tax_occur_ss1154)
```

In most of the samples, we can identify between 600 and 700 OTUs:

```{r richness_otu_no4, echo=TRUE}
plot(sort(OTUs_per_sample_18S_tax_occur_ss1154), pch=19)
boxplot(OTUs_per_sample_18S_tax_occur_ss1154, pch=19)
```

### 1.3.3) Index of evenness

#### 1.3.3.1) Pielou's index

```{r pielou_index_of_evenness1, echo=TRUE}
pielou_evenness_18S_tax_occur_ss1154<-tb18_tax_occur_ss1154_div/log(OTUs_per_sample_18S_tax_occur_ss1154)
```

The Pielou index (constrained between 0 and 1) takes values closer to 1 as the variation of species proportion in a sample increases. Most of the samples get values around 0.95, meaning that the numerical composition of different OTUs in a sample is highly variable - there's no constant presence of dominant species.

The less variation in communities between the species (and the presence of a dominant specie), the lower J' is. 

```{r pielou_index_of_evenness2, echo=TRUE}
plot(sort(pielou_evenness_18S_tax_occur_ss1154), pch=19)
boxplot(pielou_evenness_18S_tax_occur_ss1154, pch=19)
```

The OTU_38, with 890 reads, is the most abundant in the overall dataset:

```{r OTUs_overall_abundance, echo=TRUE}
head(sort(colSums(tb18_tax_occur_ss1154_no_cero), decreasing=T), n=10L)
```

Most of the OTUs show very few occurrences; suggesting that we will probably be able to identify a significant ammount of rare otus:

```{r OTUs_overall_abundance2, echo=TRUE}
plot(log(sort(colSums(tb18_tax_occur_ss1154_no_cero), decreasing=T)), pch=19)
```

<!---
#### 1.3.3.2) Sads
library(sads)
?sads
--->

### 1.3.4) Abundance Models
#### 1.3.4.1) Rank-Abundance or Dominance/Diversity Model ("radfit")

The OTUs abundance distribution fits relativelly close to log-normal model. 

```{r radfit, echo=FALSE}
#?radfit
#otu_tb18_t[1:5,1:5]

tb18_tax_occur_min1154_radfit<-radfit(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_min1154_radfit)
```

#### 1.3.4.2) Preston's Lognormal Model

According to Preston's lognormal model fit into species frequencies groups, we're missing 1588.816 species:

```{r preston_model1, echo=T}
tb18_tax_occur_ss1154_prestonfit<-prestonfit(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_ss1154_prestonfit, main="Pooled species")

veiledspec(tb18_tax_occur_ss1154_prestonfit)
```

When computing Prestons' lognormal model fit without pooling data into groups, we seem to miss 1385.371 species:

```{r preston_model2, echo=4}
tb18_tax_occur_ss1154_dist_all<-prestondistr(colSums(tb18_tax_occur_min1154_t))
plot(tb18_tax_occur_ss1154_prestonfit, main="All malaspina")
lines(tb18_tax_occur_ss1154_dist_all, line.col="blue3")

veiledspec(tb18_tax_occur_ss1154_dist_all)
```


### 1.3.5) Rarefaction curves of rarefied and non-rarefied datasets

```{r rarefraction_curve, echo=TRUE}
rarec_input<-t(as.matrix(colSums(tb18_tax_occur_ss1154_no_cero)))

#tb18_rarecurve_step1000_10000<-rarecurve(rarec_input, step = 1000, 10000, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity step=1000 & ss=10000\n(40,816 OTUs; 5,247,375 reads)\n")
#tb18_rarecurve_step1000_10000

tb18_rarecurve_step100_5700<-rarecurve(rarec_input, step = 100, 5700, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity step=100 & ss=5700\n(5,719 OTUs; 49,622 reads)\n")


#without rarefying
rarec_allOTUs_input<-t(as.matrix(colSums(t(tb18_tax_occur))))
tb18_rarecurve_allOTUs_step100_8642<-rarecurve(rarec_allOTUs_input, step = 100, 8642, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="18S amplicons diversity non-rarefied step=1000 & ss=100000\n(8,642 OTUs; 244,184 reads)\n")
```


### 1.3.6) Beta diversity

#### 1.3.6.1) Dissimilarity matrix using Bray-Curtis index:

The Bray-Curtis dissimilarity, constrained between 0 (minimum distance) and 1 (highest dissimilarity) allows us to quantify the differences between samples according to the composition and relative abundance of their OTUs. In our dataset, most of the samples pairs take dissimilarity values around 0.8, meaning that their composition is substantially different.

```{r beta_div1, echo=FALSE}
#?vegdist
tb18_tax_occur_ss1154_no_cero.bray<-vegdist(tb18_tax_occur_ss1154_no_cero, method="bray")
boxplot(tb18_tax_occur_ss1154_no_cero.bray, main="Bray-Curtis dissimilarity matrix")
```

#### 1.3.6.2) Hierarchical clustering

The stations seem to form clusters according to geographic localization, but there are no evident clusters separated from the general groups, apart from the one containing TARA_84, TARA_82 and TARA_85.

(To be done: assign Longhurst provinces information to each station and check if any of the central clusters is meaningful regarding to the samples' geographical location)

```{r beta_div2, echo=FALSE}
#UPGMA
tb18_tax_occur_ss1154_no_cero.upgma<-hclust(tb18_tax_occur_ss1154_no_cero.bray, "average")
plot(tb18_tax_occur_ss1154_no_cero.upgma, cex=.35, main="Samples Hierarchical Clustering")
```

#### 1.3.6.3) Non-metric multidimensional scaling

We can identify a prominent group in the central part of the NMDS plot and a few outliers (TARA 82, 84 and 85) in the central-right edge of the plot. The stress parameter takes a value below 0.2, suggesting that the plot is acceptable. 

```{r monoNMDS, echo=F}
#NMDS
tb18_tax_occur_ss1154_no_cero.nmds<-monoMDS(tb18_tax_occur_ss1154_no_cero.bray)
tb18_tax_occur_ss1154_no_cero.nmds
plot(tb18_tax_occur_ss1154_no_cero.nmds, main="monoMDs method")
```

When implementing a most robut function for computing NMDS plots, the result is quiet the same:

```{r metaNMDS, echo=F}
tb18_tax_occur_ss1154_no_cero.meta_nmds<-metaMDS(tb18_tax_occur_ss1154_no_cero.bray)
plot(tb18_tax_occur_ss1154_no_cero.meta_nmds, main="metaMDS method")
```

## 1.4) Geographical analysis

```{r load_geo_data, echo=F, results="hide", message=F}
#load geographical ubication of stations and sort according to otu_tb18 stations sequence.
MP_geo_18S<-read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/Taraspina_SUR_loc_nodot_stand.name2.txt", sep="\t", header=T)

MP_geo_18S<-data.table(MP_geo_18S)
#MP_geo_18S[,st_name:=(paste("st",MP_geo_18S$station,"_MD",MP_geo_18S$code,sep=""))]
MP_geo_18S<-MP_geo_18S[mixedorder(MP_geo_18S$sample),]
tb18_tax_occur_ss1154_no_cero<-tb18_tax_occur_ss1154_no_cero[order(row.names(tb18_tax_occur_ss1154_no_cero)),]
#select only stations apearing in tb18_tax_occur_ss1154_no_cero:
MP_geo_18S<-MP_geo_18S[MP_geo_18S$sample %in% row.names(tb18_tax_occur_ss1154_no_cero),]
row.names(MP_geo_18S)<-MP_geo_18S$sample

dim(MP_geo_18S)
MP_geo_18S[1:5,1:3]
tb18_tax_occur_ss1154_no_cero[1:5,1:5]

#we'll use a function from library(fossil) to read lat-long in decimal degrees and translate into distance in km. We'll print only columns containing info about station, latitude and longitude:
MP_geo_18S_reduced<-create.lats(MP_geo_18S, loc="sample", long="long", lat="lat")
head(MP_geo_18S_reduced)

#create a distance matrix (lower triangle) between a list of points.
geo_distances_MP_18S<-earth.dist(MP_geo_18S_reduced, dist = TRUE)
head(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

geo_distances_MP_18S<-as.matrix(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

#geo distances dataset ready to use: "geo_distances_MP_18S"
```

Working datasets:

1) Community matrix: tb18_tax_occur_ss1154_no_cero

```{r working_datasets1, echo=T}
dim(tb18_tax_occur_ss1154_no_cero)
tb18_tax_occur_ss1154_no_cero[1:5, 1:5]
```

2) Community Bray-Curtis: tb18_tax_occur_ss1154_no_cero.bray

```{r working_datasets2, echo=2}
tb18_tax_occur_ss1154_no_cero.bray<-vegdist(tb18_tax_occur_ss1154_no_cero, method="bray")
tb18_tax_occur_ss1154_no_cero.bray<-as.matrix(tb18_tax_occur_ss1154_no_cero.bray)
dim(tb18_tax_occur_ss1154_no_cero.bray)
```

3) Stations distances in km: geo_distances_MP_18S

```{r working_datasets3, echo=T}
dim(geo_distances_MP_18S)
```

Communities quickly change their composition across geographical distances:

```{r working_datasets4, echo=T}
plot(geo_distances_MP_18S, tb18_tax_occur_ss1154_no_cero.bray, pch=19, cex=0.4, xlab="Geopgraphical distances", ylab="Bray-Curtis dissimilarities")
```

### 1.4.1) Mantel correlograms

Mantel statistic is -significantlly- so low, meaning that the correlation between samples dissimilarity and geographical distances is weak.

```{r mantel_correlogram1, echo=T}
mantel(geo_distances_MP_18S, tb18_tax_occur_ss1154_no_cero.bray)
```

Maximum distance between samples:

```{r mantel_correlogram2, echo=F}
max(geo_distances_MP_18S)
```

Minimum distance between samples:

```{r mantel_correlogram3, echo=F}
min(geo_distances_MP_18S)
```

Correlograms:

```{r mantel_correlogram4, echo=T}
MP_18s_ss1154_mantel_correl_by_1000km<-mantel.correlog(tb18_tax_occur_ss1154_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=1000))
plot(MP_18s_ss1154_mantel_correl_by_1000km)

MP_18s_ss1154_mantel_correl_by_100km<-mantel.correlog(tb18_tax_occur_ss1154_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=100))
plot(MP_18s_ss1154_mantel_correl_by_100km)
```

## 1.5) Abundance vs. occurence

```{r OTUs_mean_relative_abund, echo=F, results="hide"}
tb18_tax_occur_ss1154_no_cero[1:5,1:5]
tb18_tax_occur_ss1154_no_cero_t<-t(tb18_tax_occur_ss1154_no_cero)

colSums(tb18_tax_occur_ss1154_no_cero_t)

#local abundance percentage
tb18_tax_occur_ss1154_no_cero_t.rabund<-tb18_tax_occur_ss1154_no_cero_t/1154

colSums(tb18_tax_occur_ss1154_no_cero_t.rabund)
tb18_tax_occur_ss1154_no_cero_t.rabund[1:5,1:5]

#OTUs mean relative abundance
tb18_tax_occur_ss1154_no_cero_t.rabund_means<-rowMeans(tb18_tax_occur_ss1154_no_cero_t.rabund) 
tb18_tax_occur_ss1154_no_cero_t.rabund_means<-as.data.frame(tb18_tax_occur_ss1154_no_cero_t.rabund_means)

head(tb18_tax_occur_ss1154_no_cero_t.rabund_means)
```

```{r OTUs_occurence, echo=F, results='hide'}
tb18_tax_occur_ss1154_no_cero_t.rabund.occur<-tb18_tax_occur_ss1154_no_cero_t.rabund
tb18_tax_occur_ss1154_no_cero_t.rabund.occur[tb18_tax_occur_ss1154_no_cero_t.rabund.occur>0]<-1
tb18_tax_occur_ss1154_no_cero_t.rabund.occur[1:5,1:5] # presence - absence table

#percentage of occurence in overall stations
tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc<-as.data.frame(100*(rowSums(tb18_tax_occur_ss1154_no_cero_t.rabund.occur)/43))

str(tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc)
```

```{r merge_rabund_peroccur, echo=F, results='hide'}
otu_tb18_ss1154_rabund_percoccur<-merge(tb18_tax_occur_ss1154_no_cero_t.rabund_means,tb18_tax_occur_ss1154_no_cero_t.rabund_means.occurence_perc, by="row.names")

colnames(otu_tb18_ss1154_rabund_percoccur)<-c("OTUs","mean_rabund","perc_occur")
otu_tb18_ss1154_rabund_percoccur[1:5,]

row.names(otu_tb18_ss1154_rabund_percoccur)<-otu_tb18_ss1154_rabund_percoccur[,1]
otu_tb18_ss1154_rabund_percoccur<-otu_tb18_ss1154_rabund_percoccur[,-1]
otu_tb18_ss1154_rabund_percoccur[1:5,]
```

In the following plot, we can appreciate the OTUs distribution according to their percentage of occurence and relative abundance. The red line keeps up OTUs that occur in more than 80% of the samples, the green line limits regionally rare OTUs (< 0.001%), and the blue one restricts regionally abundant OTUs (> 0.1%).

```{r abund_vs_occurence_table, echo=F}
plot(otu_tb18_ss1154_rabund_percoccur$mean_rabund,otu_tb18_ss1154_rabund_percoccur$perc_occur, log="x", pch=19, cex=0.8, xlab="Mean relative abundance", ylab="Percentage of occurence")
abline(h=80, col="red") #occurence higher than 80%
abline(v=0.00001, col="green") #rare OTUs
abline(v=0.001, col="blue") #cosmopolitan OTUs

#Conventional limits:
#Regionally rare     = 0.00001
#Regionally abundant = 0.001
```

Regionally abundant OTUs (relative abundance over 0.1%):

```{r abundant_OTUs, echo=7}
#regionally abundant
tb18_ss1154_abundant<-otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund > 0.001,]

tb18_ss1154_abundant_sorted<-tb18_ss1154_abundant[order(tb18_ss1154_abundant$mean_rabund, tb18_ss1154_abundant$perc_occur, decreasing = T), c(1,2)]

tb18_ss1154_abundant_sorted
dim(tb18_ss1154_abundant_sorted)
```

Proportion of regionally abundant OTUs (%):

```{r abundant_OTUs2, echo=F}
(nrow(tb18_ss1154_abundant_sorted)/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```

Cosmopolitan OTUs (relative abundance over 0.1% and occurence in more than 80% of samples):

```{r select_cosmopolitan, echo=6}
otu_tb18_ss1154_rabund_cosm<-otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund > 0.001,]
otu_tb18_ss1154_rabund_poccur_cosm<-otu_tb18_ss1154_rabund_cosm[otu_tb18_ss1154_rabund_cosm$perc_occur > 80,]
otu_tb18_ss1154_cosmop_sorted<-otu_tb18_ss1154_rabund_poccur_cosm[order(otu_tb18_ss1154_rabund_poccur_cosm$perc_occur, otu_tb18_ss1154_rabund_poccur_cosm$mean_rabund, decreasing = T), c(1,2)]

otu_tb18_ss1154_cosmop_sorted
dim(otu_tb18_ss1154_cosmop_sorted)
```

Number and proportion (%) of cosmopolitan OTUs:

```{r percentage_cosmopolitan, echo=F}
nrow(otu_tb18_ss1154_cosmop_sorted)

(nrow(otu_tb18_ss1154_cosmop_sorted)/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```

Number and proportion (%) of rare OTUs:

```{r rare_OTUs, echo=1}
nrow(otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund < 0.00001 & otu_tb18_ss1154_rabund_percoccur$mean_rabund >0,])

(nrow(otu_tb18_ss1154_rabund_percoccur[otu_tb18_ss1154_rabund_percoccur$mean_rabund < 0.00001 & otu_tb18_ss1154_rabund_percoccur$mean_rabund >0,])/nrow(otu_tb18_ss1154_rabund_percoccur))*100
```




## 1.6) Taxonomic composition analysis

### 1.6.1) Normalized data

#### 1.6.1.1) Absolute values

Let's add the taxonomic classification by merging "tb18_tax_occur_ss1154_no_cero" with "tb18_tax":

```{r merge_tables, echo=FALSE}
tb18_tax_occur_ss1154_no_cero_t<-t(tb18_tax_occur_ss1154_no_cero)
tb18_ss1154_tax<-merge(tb18_tax_occur_ss1154_no_cero_t,tb18_class, by="row.names")

dim(tb18_ss1154_tax)
tb18_ss1154_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_ss1154_tax)=tb18_ss1154_tax$Row.names
#add OTU_no as rowname
rownames.tb18_ss1154_tax<-tb18_ss1154_tax[,1]
tb18_ss1154_tax<-tb18_ss1154_tax[,-1]

dim(tb18_ss1154_tax)
tb18_ss1154_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_ss1154_tax["OTU_no"] <- NA
tb18_ss1154_tax$OTU_no <- sapply(strsplit(rownames(tb18_ss1154_tax),split= "\\_"),'[',2)
tb18_ss1154_tax$OTU_no <- as.numeric(as.character(tb18_ss1154_tax$OTU_no))
tb18_ss1154_tax_sorted<-tb18_ss1154_tax[order(tb18_ss1154_tax$OTU_no, decreasing = FALSE), ]

dim(tb18_ss1154_tax_sorted)
tb18_ss1154_tax_sorted[1:5,1:5]
write.table(tb18_ss1154_tax_sorted, "test", sep="\t", row.names = F)
```

<!---
```{r select_phototrophs2, echo=FALSE}
# [TEMPORARY EDITION] for some reason, the taxonomic classification of "OTU_2485" is not properly parsed. As it is not classified as a phototroph anyway, by now let's get rid of it by excluding empty cells (""). 
tb18_phototrophs <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif != "" & tb18_ss1154_tax_sorted$classif != "NA"),]

dim(tb18_phototrophs)
```
--->

```{r select_phototrophs, echo=FALSE}
tb18_phototrophs <- tb18_ss1154_tax_sorted[which(tb18_ss1154_tax_sorted$classif != "NA"),]

dim(tb18_phototrophs)
```

```{r count_samples_per_group, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
#length(Dinophyceae_tb_samples[which(colSums(Dinophyceae_tb_occur) != 0)])

Prasinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Cryptophyceae"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:43]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bacillariophyceae"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:43]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorarachniophyceae"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:43]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pinguiophyceae"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:43]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Eustigmatophyceae"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:43]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])

Phaeophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])

Phaeothamniophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])

Xanthophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])

Chlorodendrophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])

IncertaeSedis_Archaeplastida_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])

Nephroselmidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])

Pavlovophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])

Rhodophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])

Rappemonads_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])

MOCH_1_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH-1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])

MOCH_2_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH-2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])

MOCH_5_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "MOCH-5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])

Prasinophyceae_clade_VII_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])

Prasinophyceae_clade_IX_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])

Pyramimonadaceae_tb <- tb18_phototrophs[which(tb18_phototrophs$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
```

```{r aggregate, echo=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_phototrophs[1:43]), list(tb18_phototrophs$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples, echo=FALSE}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples["samples_per_class"] <- NA
tb18S_OTUs_reads_samples$samples_per_class<-c(43,43,40,42,39,43,42,32,38,35,28,36,30,22,16,26,17,13,11,1,3,1,1)

tb18S_OTUs_reads_samples
```






#### 1.6.1.1) Relative values

```{r OTUs_vs_reads_relative_values_18S, echo=FALSE}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads_18S_v2, echo=FALSE}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[4.3] Reads per class vs. OTUs per class", x="\nMP-iTags-18S OTUs (%)", y="MP-iTags-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


```{r reads_vs_samples_18S, echo=FALSE}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[New] Reads per class vs. samples in which they occur", x="\nMP-iTags-18S samples (%)", y="MP-iTags-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


<!---
OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs_v1, echo=FALSE}
#library(ggplot2)
#library(ggrepel)

#otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples)))

#otus_vs_samples_plot + 
#geom_point() + 
#geom_text_repel(
# aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples),
#  mapping = TRUE,
#  inherit.aes = TRUE,
#  size = 3,
#  box.padding = unit(0.1, "lines"),
#  point.padding = unit(0.15, "lines")
#  ) +

#geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 137)) + 
#labs(title="[1] OTUs per class vs. samples in which they occur", x="samples", y="OTUs") + coord_trans(y = "log10") 
```


```{r stations_vs_OTUs, echo=FALSE}
ggplot(tb18S_OTUs_reads_samples) +
  geom_point(aes(samples_per_class,OTUs_per_class)) +
  geom_text_repel(aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=3, force = 3, box.padding = unit(0.05, "lines"), point.padding = unit(0.1, 'lines'), nudge_y = 15) + 
  scale_x_continuous(limits = c(0, 137)) + 
  labs(title="[1] OTUs per class vs. samples in which they occur", x="samples", y="OTUs") + 
  coord_trans(y = "log10") 
```


Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_log10(limits = c(-1, 15000), breaks=c(0.1,1,10,100,1000,10000)) + scale_y_continuous(trans = "log10") + labs(title="[2] Reads per class vs. OTUs per class", x="OTUs (log10)", y="reads (log10)") 


ggplot(tb18S_OTUs_reads_samples) +
  geom_point(aes(OTUs_per_class,reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_log10(limits = c(-1, 15000), breaks=c(0,1,10,100,1000,10000)) + scale_y_log10(limits = c(-1,1000000), breaks = c(0,1,10,100,1000,10000,100000,1000000)) + labs(title="[2.2] Reads per class vs. OTUs per class", x="\nMP-iTags-18S OTUs", y="MP-iTags-18S reads\n") + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))

```

OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs_18S, echo=FALSE}
#otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund, aes(samples_per_class,OTUs_per_class, #label = rownames(tb18S_OTUs_reads_samples)))
#otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = #3) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[3] OTUs per class vs. #samples in which they occur", x="MP-iTags-18S samples (%)", y="OTUs (%)") + coord_trans(y = "log10")


ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,OTUs_per_class)) +
  geom_text_repel(aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 5, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[3.2] OTUs per class vs. samples in which they occur", x="\nMP-iTags-18S samples (%)", y="MP-iTags-18S OTUs (%)\n") + coord_trans(y = "log10") + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + labs(title="[4] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1,200), breaks = c(0.1,1,10,100))


ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 0.5, box.padding = unit(0.2, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[4.2] Reads per class vs. OTUs per class", x="\nMP-iTags-18S OTUs (%)", y="MP-iTags-18S reads (%)\n") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1,200), breaks = c(0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```
--->

### 1.6.2) Non-normalized data

```{r merge_tables_non.norm_18S, echo=FALSE}
tb18_tax_occur_min1154_t<-tb18_tax_occur_min1154
tb18_min1154_tax<-merge(tb18_tax_occur_min1154_t,tb18_class, by="row.names")

dim(tb18_min1154_tax)
tb18_min1154_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_min1154_tax)=tb18_min1154_tax$Row.names

#add OTU_no as rowname
rownames.tb18_min1154_tax<-tb18_min1154_tax[,1]
tb18_min1154_tax<-tb18_min1154_tax[,-1]
#colnames(tb18_min1154_tax, do.NULL=F)

dim(tb18_min1154_tax)
tb18_min1154_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_min1154_tax["OTU_no"] <- NA
tb18_min1154_tax$OTU_no <- sapply(strsplit(rownames(tb18_min1154_tax),split= "\\_"),'[',2)
tb18_min1154_tax$OTU_no <- as.numeric(as.character(tb18_min1154_tax$OTU_no))
tb18_min1154_tax_sorted<-tb18_min1154_tax[order(tb18_min1154_tax$OTU_no, decreasing = FALSE), ]

dim(tb18_min1154_tax_sorted)
tb18_min1154_tax_sorted[1:5,1:5]
```

```{r select_phototrophs_non.norm_18S, echo=FALSE}
tb18_phototrophs_non.norm <- tb18_min1154_tax_sorted[which(tb18_min1154_tax_sorted$classif != "" & tb18_min1154_tax_sorted$classif != "NA"),]

dim(tb18_phototrophs_non.norm)
```

```{r count_samples_per_group_non.norm_18S, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:43]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
#length(Dinophyceae_tb_samples[which(colSums(Dinophyceae_tb_occur) != 0)])

Prasinophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "other_Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:43]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:43]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:43]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:43]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Cryptophyceae"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:43]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Bacillariophyceae"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:43]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Chlorarachniophyceae"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:43]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:43]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Pinguiophyceae"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:43]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:43]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:43]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Eustigmatophyceae"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:43]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:43]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:43]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:43]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:43]
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])

Phaeophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Phaeophyceae"),]
Phaeophyceae_tb_occur <- Phaeophyceae_tb[,1:43]
length(Phaeophyceae_tb_occur[,colSums(Phaeophyceae_tb_occur) > 0])

Phaeothamniophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:43]
length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])

Xanthophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:43]
length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])

Chlorodendrophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:43]
length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])

IncertaeSedis_Archaeplastida_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "IncertaeSedis_Archaeplastida"),]
IncertaeSedis_Archaeplastida_tb_occur <- IncertaeSedis_Archaeplastida_tb[,1:43]
length(IncertaeSedis_Archaeplastida_tb_occur[,colSums(IncertaeSedis_Archaeplastida_tb_occur) > 0])

Nephroselmidophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Nephroselmidophyceae"),]
Nephroselmidophyceae_tb_occur <- Nephroselmidophyceae_tb[,1:43]
length(Nephroselmidophyceae_tb_occur[,colSums(Nephroselmidophyceae_tb_occur) > 0])

Pavlovophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:43]
length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])

Rhodophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Rhodophyceae"),]
Rhodophyceae_tb_occur <- Rhodophyceae_tb[,1:43]
length(Rhodophyceae_tb_occur[,colSums(Rhodophyceae_tb_occur) > 0])

Rappemonads_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Rappemonads"),]
Rappemonads_tb_occur <- Rappemonads_tb[,1:43]
length(Rappemonads_tb_occur[,colSums(Rappemonads_tb_occur) > 0])

MOCH_1_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "MOCH-1"),]
MOCH_1_tb_occur <- MOCH_1_tb[,1:43]
length(MOCH_1_tb_occur[,colSums(MOCH_1_tb_occur) > 0])

MOCH_2_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "MOCH-2"),]
MOCH_2_tb_occur <- MOCH_2_tb[,1:43]
length(MOCH_2_tb_occur[,colSums(MOCH_2_tb_occur) > 0])

MOCH_5_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "MOCH-5"),]
MOCH_5_tb_occur <- MOCH_5_tb[,1:43]
length(MOCH_5_tb_occur[,colSums(MOCH_5_tb_occur) > 0])

Prasinophyceae_clade_VII_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Prasinophyceae_clade-VII"),]
Prasinophyceae_clade_VII_tb_occur <- Prasinophyceae_clade_VII_tb[,1:43]
length(Prasinophyceae_clade_VII_tb_occur[,colSums(Prasinophyceae_clade_VII_tb_occur) > 0])

Prasinophyceae_clade_IX_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Prasinophyceae_clade-IX"),]
Prasinophyceae_clade_IX_tb_occur <- Prasinophyceae_clade_IX_tb[,1:43]
length(Prasinophyceae_clade_IX_tb_occur[,colSums(Prasinophyceae_clade_IX_tb_occur) > 0])

Pyramimonadaceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$classif == "Pyramimonadaceae"),]
Pyramimonadaceae_tb_occur <- Pyramimonadaceae_tb[,1:43]
length(Pyramimonadaceae_tb_occur[,colSums(Pyramimonadaceae_tb_occur) > 0])
```

```{r aggregate_non.norm_18S, echo=FALSE}
class_summary_reads_per_class_non.norm<-aggregate(rowSums(tb18_phototrophs_non.norm[1:43]), list(tb18_phototrophs_non.norm$classif), sum)
# count the different groups

class_summary_otus_per_class_non.norm<-aggregate(rowSums(tb18_phototrophs_non.norm[1:43]), list(tb18_phototrophs_non.norm$classif), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_non.norm)
class_summary_reads_per_class_non.norm_order<-class_summary_reads_per_class_non.norm[order(-x),]
detach(class_summary_reads_per_class_non.norm)
class_summary_reads_per_class_non.norm_order

#fix column names
row.names(class_summary_reads_per_class_non.norm_order)<-class_summary_reads_per_class_non.norm_order[,1]
class_summary_reads_per_class_non.norm_order<-class_summary_reads_per_class_non.norm_order[c(-1)]
colnames(class_summary_reads_per_class_non.norm_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class_non.norm)
class_summary_otus_per_class_non.norm_order<-class_summary_otus_per_class_non.norm[order(-x),]
detach(class_summary_otus_per_class_non.norm)
class_summary_otus_per_class_non.norm_order

row.names(class_summary_otus_per_class_non.norm_order)<-class_summary_otus_per_class_non.norm_order[,1]
class_summary_otus_per_class_non.norm_order<-class_summary_otus_per_class_non.norm_order[c(-1)]
colnames(class_summary_otus_per_class_non.norm_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_non.norm_18S, echo=FALSE}
tb18S_OTUs_reads_samples_non.norm <- merge(class_summary_reads_per_class_non.norm_order, class_summary_otus_per_class_non.norm_order, by="row.names")

row.names(tb18S_OTUs_reads_samples_non.norm)<-tb18S_OTUs_reads_samples_non.norm[,1]

tb18S_OTUs_reads_samples_non.norm<-tb18S_OTUs_reads_samples_non.norm[,-1]
colnames(tb18S_OTUs_reads_samples_non.norm)<-c("reads_per_class","OTUs_per_class")
tb18S_OTUs_reads_samples_non.norm[1:5,1:2]

tb18S_OTUs_reads_samples_non.norm<-tb18S_OTUs_reads_samples_non.norm[order(tb18S_OTUs_reads_samples_non.norm$reads_per_class, tb18S_OTUs_reads_samples_non.norm$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples_non.norm["samples_per_class"] <- NA
tb18S_OTUs_reads_samples_non.norm$samples_per_class<-c(43,43,43,113,116,103,112,112,108,111,97,65,35,75,75,50,18,1,1,1,1,1,1,1,1)

tb18S_OTUs_reads_samples_non.norm
```

```{r OTUs_vs_reads_relative_values_non.norm_18S, echo=FALSE}
tb18S_OTUs_reads_samples_rel_abund_non.norm <- tb18S_OTUs_reads_samples_non.norm

tb18S_OTUs_reads_samples_rel_abund_non.norm$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples_non.norm)[1]
tb18S_OTUs_reads_samples_rel_abund_non.norm$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples_non.norm)[2]
tb18S_OTUs_reads_samples_rel_abund_non.norm$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$samples_per_class*100)/43

colSums(tb18S_OTUs_reads_samples_rel_abund_non.norm)
tb18S_OTUs_reads_samples_rel_abund_non.norm
```

OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs_non.norm_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund_non.norm, aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund_non.norm)))
otus_vs_samples_plot + 
  geom_point() + 
  geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + 
  scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + 
  labs(title="[5] OTUs per class vs. samples in which they occur", x="MP-iTags-18S samples (%)", y="OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads_non.norm_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund_non.norm, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund_non.norm)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + labs(title="[6] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1, 200), breaks=c(0.1,1,10,100))
```



############################




# 2) 16S amplicons

## 2.1) Data overview

Let's read the dataset and remove the samples containing less than 36155 reads:

``` {r load_data_16S, echo=FALSE, message=FALSE}

#read data 
tb16_tax <- read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/otu_table99_SUR_miTags_16S_classif.txt", head=TRUE, fill=TRUE, sep="\t")

#table dimensions and format before setting column names
dim(tb16_tax) # 28294    59
tb16_tax[1:5,1:5]

row.names(tb16_tax)<-tb16_tax[,1]
tb16_tax<-tb16_tax[,-1]
tb16_tax[is.na(tb16_tax)]<-"NA"

dim(tb16_tax) #28294    58
tb16_tax[1:5,1:5]

#table with taxonomic classification alone:
tb16_class <- tb16_tax[,56:58]
dim(tb16_class) #28294     3
tb16_class[1:5,1:3]

#table with occurence data alone
tb16_tax_occur <- tb16_tax[,1:55]
dim(tb16_tax_occur) #28294    55
tb16_tax_occur[1:5,1:5]

amplicons_per_sample_tb16<-colSums(tb16_tax_occur)
summary(amplicons_per_sample_tb16) #median of amplicons per sample = ~83020)
#we'll remove all samples with less than 36155 reads.
sort(amplicons_per_sample_tb16[which(colSums(tb16_tax_occur)<36155)])

#remove samples with less than 36155 reads
tb16_tax_occur_min36155 <- tb16_tax_occur[,colSums(tb16_tax_occur) >= 36155]
dim(tb16_tax_occur_min36155) #959 107

#remove samples with omitted in 18S dataset (so that we can compare the relative abundance of 18S and 16S OTUs considering the same samples)
tb16_tax_occur_min36155<-subset(tb16_tax_occur_min36155, select=-c(TARA_100_SUR_0d2_3,TARA_141_SUR_0d2_3,TARA_93_SUR_0d2_3,TARA_94_SUR_0d2_3,MP0528,MP0778))

dim(tb16_tax_occur_min36155) #28294    44

#let's check if we loose any OTU when excluding the mentioned stations.
tb16_tax_occur_min36155_no_cero<-tb16_tax_occur_min36155[,-(which(colSums(tb16_tax_occur_min36155)==0))]
dim(tb16_tax_occur_min36155_no_cero)
```


Table dimensions and content outline:

```{r starting_dataset_16S, echo=FALSE}
dim(tb16_tax_occur_min36155)
tb16_tax_occur_min36155[1:5,1:5]
```

Minimum number of reads per station:

```{r reads_per_sample_overview1_16S, echo=1}
min(colSums(tb16_tax_occur_min36155)) 
#36155
```

Maximum number of reads per station:

```{r reads_per_sample_overview2_16S, echo=1}
max(colSums(tb16_tax_occur_min36155)) 
# max: 935755
```

Identification of stations with higher number of reads:

```{r reads_per_sample_overview3_16S, echo=TRUE}
amplicons_per_sample<-colSums(tb16_tax_occur_min36155)
amplicons_per_sample[which(colSums(tb16_tax_occur_min36155)>150000)]
```

Overall reads per sample:

``` {r reads_per_sample_overview4_16S, echo=FALSE}
plot(sort(colSums(tb16_tax_occur_min36155)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 2.2) Normalization

Let's normalize the original dataset by randomly subsampling 36155 reads in each station:

``` {r species_richness_rarefaction1_16S, echo=TRUE}
tb16_tax_occur_min36155_t<-t(tb16_tax_occur_min36155)
tb16_tax_occur_ss36155<-rrarefy(tb16_tax_occur_min36155_t, 36155)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2_16S, echo=FALSE}
dim(tb16_tax_occur_ss36155)
tb16_tax_occur_ss36155[1:5,1:5]
```

Its content fits with the expected normalization values (36155 reads per station):

``` {r species_richness_rarefaction3_16S, echo=TRUE}
rowSums(tb16_tax_occur_ss36155)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4_16S, echo=1:5}
length(which(colSums(tb16_tax_occur_ss36155)==0)) #46
```

There are 8045 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5_16S, echo=1:3}
tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155[,-(which(colSums(tb16_tax_occur_ss36155)==0))]
tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155_no_cero[mixedorder(row.names(tb16_tax_occur_ss36155_no_cero)),]
dim(tb16_tax_occur_ss36155_no_cero) #92 913

#the final dimensions of the normalized table are 43 26657.
#913 + 46 = 959
```

Datasets summary:
```{r dataset_summary_16S}
dim(tb16_tax) #959 128
dim(tb16_tax_occur) #959 122
dim(tb16_tax_occur_ss36155_no_cero) #92 913
```



## 2.3) General community analysis

### 2.3.1) Richness and evenness (Shannon index)

```{r shannon_index1, echo=FALSE}
tb16_tax_occur_ss36155_div <- diversity(tb16_tax_occur_ss36155_no_cero, index="shannon")
```

Most of the samples take Shannon Index values around 6:

```{r shannon_index2, echo=FALSE}
boxplot(tb16_tax_occur_ss36155_div, pch=19, main="Shannon's index of diversity")
plot(sort(tb16_tax_occur_ss36155_div), pch=19, main="Shannon's index of diversity")
```

### 1.3.2) Richness: OTU number

```{r richness_otu_no1, echo=FALSE}
OTUs_per_sample_16S_tax_occur_ss36155<-specnumber(tb16_tax_occur_ss36155_no_cero)
```

Lowest number of OTUs per sample:

```{r richness_otu_no2, echo=FALSE}
min(OTUs_per_sample_16S_tax_occur_ss36155)
```

Maximum number of OTUs per sample:

```{r richness_otu_no3, echo=FALSE}
max(OTUs_per_sample_16S_tax_occur_ss36155)
```

In most of the samples, we can identify between 600 and 700 OTUs:

```{r richness_otu_no4, echo=TRUE}
plot(sort(OTUs_per_sample_16S_tax_occur_ss36155), pch=19)
boxplot(OTUs_per_sample_16S_tax_occur_ss36155, pch=19)
```

### 2.3.3) Index of evenness

#### 2.3.3.1) Pielou's index

```{r pielou_index_of_evenness1, echo=TRUE}
pielou_evenness_16S_tax_occur_ss36155<-tb16_tax_occur_ss36155_div/log(OTUs_per_sample_16S_tax_occur_ss36155)
```

The Pielou index (constrained between 0 and 1) takes values closer to 1 as the variation of species proportion in a sample increases. Most of the samples get values around 0.95, meaning that the numerical composition of different OTUs in a sample is highly variable - there's no constant presence of dominant species.

The less variation in communities between the species (and the presence of a dominant specie), the lower J' is. 

```{r pielou_index_of_evenness2, echo=TRUE}
plot(sort(pielou_evenness_16S_tax_occur_ss36155), pch=19)
boxplot(pielou_evenness_16S_tax_occur_ss36155, pch=19)
```

The OTU_38, with 890 reads, is the most abundant in the overall dataset:

```{r OTUs_overall_abundance, echo=TRUE}
head(sort(colSums(tb16_tax_occur_ss36155_no_cero), decreasing=T), n=10L)
```

Most of the OTUs show very few occurrences; suggesting that we will probably be able to identify a significant ammount of rare otus:

```{r OTUs_overall_abundance2, echo=TRUE}
plot(log(sort(colSums(tb16_tax_occur_ss36155_no_cero), decreasing=T)), pch=19)
```

<!---
#### 1.3.3.2) Sads
library(sads)
?sads
--->

### 1.3.4) Abundance Models
#### 1.3.4.1) Rank-Abundance or Dominance/Diversity Model ("radfit")

The OTUs abundance distribution fits relativelly close to log-normal model. 

```{r radfit, echo=FALSE}
#?radfit
#otu_tb16_t[1:5,1:5]

tb16_tax_occur_min36155_radfit<-radfit(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_min36155_radfit)
```

#### 1.3.4.2) Preston's Lognormal Model

According to Preston's lognormal model fit into species frequencies groups, we're missing 1588.816 species:

```{r preston_model1, echo=T}
tb16_tax_occur_ss36155_prestonfit<-prestonfit(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_ss36155_prestonfit, main="Pooled species")

veiledspec(tb16_tax_occur_ss36155_prestonfit)
```

When computing Prestons' lognormal model fit without pooling data into groups, we seem to miss 1385.371 species:

```{r preston_model2, echo=4}
tb16_tax_occur_ss36155_dist_all<-prestondistr(colSums(tb16_tax_occur_min36155_t))
plot(tb16_tax_occur_ss36155_prestonfit, main="All malaspina")
lines(tb16_tax_occur_ss36155_dist_all, line.col="blue3")

veiledspec(tb16_tax_occur_ss36155_dist_all)
```


### 1.3.5) Rarefaction curves of rarefied and non-rarefied datasets

```{r rarefraction_curve, echo=TRUE}
rarec_input<-t(as.matrix(colSums(tb16_tax_occur_ss36155_no_cero)))

#tb16_rarecurve_step1000_10000<-rarecurve(rarec_input, step = 1000, 10000, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="16S amplicons diversity step=1000 & ss=10000\n(40,816 OTUs; 5,247,375 reads)\n")
#tb16_rarecurve_step1000_10000

tb16_rarecurve_step100_5700<-rarecurve(rarec_input, step = 100, 5700, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="16S amplicons diversity step=100 & ss=5700\n(5,719 OTUs; 49,622 reads)\n")


#without rarefying
rarec_allOTUs_input<-t(as.matrix(colSums(t(tb16_tax_occur))))
tb16_rarecurve_allOTUs_step100_8642<-rarecurve(rarec_allOTUs_input, step = 100, 8642, xlab = "Sample size", ylab = "OTUs", label = TRUE, main="16S amplicons diversity non-rarefied step=1000 & ss=100000\n(8,642 OTUs; 244,164 reads)\n")
```


### 1.3.6) Beta diversity

#### 1.3.6.1) Dissimilarity matrix using Bray-Curtis index:

The Bray-Curtis dissimilarity, constrained between 0 (minimum distance) and 1 (highest dissimilarity) allows us to quantify the differences between samples according to the composition and relative abundance of their OTUs. In our dataset, most of the samples pairs take dissimilarity values around 0.8, meaning that their composition is substantially different.

```{r beta_div1, echo=FALSE}
#?vegdist
tb16_tax_occur_ss36155_no_cero.bray<-vegdist(tb16_tax_occur_ss36155_no_cero, method="bray")
boxplot(tb16_tax_occur_ss36155_no_cero.bray, main="Bray-Curtis dissimilarity matrix")
```

#### 1.3.6.2) Hierarchical clustering

The stations seem to form clusters according to geographic localization, but there are no evident clusters separated from the general groups, apart from the one containing TARA_84, TARA_82 and TARA_85.

(To be done: assign Longhurst provinces information to each station and check if any of the central clusters is meaningful regarding to the samples' geographical location)

```{r beta_div2, echo=FALSE}
#UPGMA
tb16_tax_occur_ss36155_no_cero.upgma<-hclust(tb16_tax_occur_ss36155_no_cero.bray, "average")
plot(tb16_tax_occur_ss36155_no_cero.upgma, cex=.35, main="Samples Hierarchical Clustering")
```

#### 1.3.6.3) Non-metric multidimensional scaling

We can identify a prominent group in the central part of the NMDS plot and a few outliers (TARA 82, 84 and 85) in the central-right edge of the plot. The stress parameter takes a value below 0.2, suggesting that the plot is acceptable. 

```{r monoNMDS, echo=F}
#NMDS
tb16_tax_occur_ss36155_no_cero.nmds<-monoMDS(tb16_tax_occur_ss36155_no_cero.bray)
tb16_tax_occur_ss36155_no_cero.nmds
plot(tb16_tax_occur_ss36155_no_cero.nmds, main="monoMDs method")
```

When implementing a most robut function for computing NMDS plots, the result is quiet the same:

```{r metaNMDS, echo=F}
tb16_tax_occur_ss36155_no_cero.meta_nmds<-metaMDS(tb16_tax_occur_ss36155_no_cero.bray)
plot(tb16_tax_occur_ss36155_no_cero.meta_nmds, main="metaMDS method")
```

## 1.4) Geographical analysis

```{r load_geo_data, echo=F, results="hide", message=F}
#load geographical ubication of stations and sort according to otu_tb16 stations sequence.
MP_geo_16S<-read.table(file="~/Documents/TFM2/TFM2_Taraspina_miTags_analysis/input/Taraspina_SUR_loc_nodot_stand.name2.txt", sep="\t", header=T)

MP_geo_16S<-data.table(MP_geo_16S)
#MP_geo_16S[,st_name:=(paste("st",MP_geo_16S$station,"_MD",MP_geo_16S$code,sep=""))]
MP_geo_16S<-MP_geo_16S[mixedorder(MP_geo_16S$sample),]
tb16_tax_occur_ss36155_no_cero<-tb16_tax_occur_ss36155_no_cero[order(row.names(tb16_tax_occur_ss36155_no_cero)),]
#select only stations apearing in tb16_tax_occur_ss36155_no_cero:
MP_geo_16S<-MP_geo_16S[MP_geo_16S$sample %in% row.names(tb16_tax_occur_ss36155_no_cero),]
row.names(MP_geo_16S)<-MP_geo_16S$sample

dim(MP_geo_16S)
MP_geo_16S[1:5,1:3]
tb16_tax_occur_ss36155_no_cero[1:5,1:5]

#we'll use a function from library(fossil) to read lat-long in decimal degrees and translate into distance in km. We'll print only columns containing info about station, latitude and longitude:
MP_geo_16S_reduced<-create.lats(MP_geo_16S, loc="sample", long="long", lat="lat")
head(MP_geo_16S_reduced)

#create a distance matrix (lower triangle) between a list of points.
geo_distances_MP_16S<-earth.dist(MP_geo_16S_reduced, dist = TRUE)
head(geo_distances_MP_16S)
dim(geo_distances_MP_16S)

geo_distances_MP_16S<-as.matrix(geo_distances_MP_16S)
dim(geo_distances_MP_16S)

#geo distances dataset ready to use: "geo_distances_MP_16S"
```

Working datasets:

1) Community matrix: tb16_tax_occur_ss36155_no_cero

```{r working_datasets1, echo=T}
dim(tb16_tax_occur_ss36155_no_cero)
tb16_tax_occur_ss36155_no_cero[1:5, 1:5]
```

2) Community Bray-Curtis: tb16_tax_occur_ss36155_no_cero.bray

```{r working_datasets2, echo=2}
tb16_tax_occur_ss36155_no_cero.bray<-vegdist(tb16_tax_occur_ss36155_no_cero, method="bray")
tb16_tax_occur_ss36155_no_cero.bray<-as.matrix(tb16_tax_occur_ss36155_no_cero.bray)
dim(tb16_tax_occur_ss36155_no_cero.bray)
```

3) Stations distances in km: geo_distances_MP_16S

```{r working_datasets3, echo=T}
dim(geo_distances_MP_16S)
```

Communities quickly change their composition across geographical distances:

```{r working_datasets4, echo=T}
plot(geo_distances_MP_16S, tb16_tax_occur_ss36155_no_cero.bray, pch=19, cex=0.4, xlab="Geopgraphical distances", ylab="Bray-Curtis dissimilarities")
```

### 1.4.1) Mantel correlograms

Mantel statistic is -significantlly- so low, meaning that the correlation between samples dissimilarity and geographical distances is weak.

```{r mantel_correlogram1, echo=T}
mantel(geo_distances_MP_16S, tb16_tax_occur_ss36155_no_cero.bray)
```

Maximum distance between samples:

```{r mantel_correlogram2, echo=F}
max(geo_distances_MP_16S)
```

Minimum distance between samples:

```{r mantel_correlogram3, echo=F}
min(geo_distances_MP_16S)
```

Correlograms:

```{r mantel_correlogram4, echo=T}
MP_16s_ss36155_mantel_correl_by_1000km<-mantel.correlog(tb16_tax_occur_ss36155_no_cero.bray, D.geo=geo_distances_MP_16S, break.pts=seq(0,20000, by=1000))
plot(MP_16s_ss36155_mantel_correl_by_1000km)

MP_16s_ss36155_mantel_correl_by_100km<-mantel.correlog(tb16_tax_occur_ss36155_no_cero.bray, D.geo=geo_distances_MP_16S, break.pts=seq(0,20000, by=100))
plot(MP_16s_ss36155_mantel_correl_by_100km)
```

## 1.5) Abundance vs. occurence

```{r OTUs_mean_relative_abund, echo=F, results="hide"}
tb16_tax_occur_ss36155_no_cero[1:5,1:5]
tb16_tax_occur_ss36155_no_cero_t<-t(tb16_tax_occur_ss36155_no_cero)

colSums(tb16_tax_occur_ss36155_no_cero_t)

#local abundance percentage
tb16_tax_occur_ss36155_no_cero_t.rabund<-tb16_tax_occur_ss36155_no_cero_t/36155

colSums(tb16_tax_occur_ss36155_no_cero_t.rabund)
tb16_tax_occur_ss36155_no_cero_t.rabund[1:5,1:5]

#OTUs mean relative abundance
tb16_tax_occur_ss36155_no_cero_t.rabund_means<-rowMeans(tb16_tax_occur_ss36155_no_cero_t.rabund) 
tb16_tax_occur_ss36155_no_cero_t.rabund_means<-as.data.frame(tb16_tax_occur_ss36155_no_cero_t.rabund_means)

head(tb16_tax_occur_ss36155_no_cero_t.rabund_means)
```

```{r OTUs_occurence, echo=F, results='hide'}
tb16_tax_occur_ss36155_no_cero_t.rabund.occur<-tb16_tax_occur_ss36155_no_cero_t.rabund
tb16_tax_occur_ss36155_no_cero_t.rabund.occur[tb16_tax_occur_ss36155_no_cero_t.rabund.occur>0]<-1
tb16_tax_occur_ss36155_no_cero_t.rabund.occur[1:5,1:5] # presence - absence table

#percentage of occurence in overall stations
tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc<-as.data.frame(100*(rowSums(tb16_tax_occur_ss36155_no_cero_t.rabund.occur)/43))

str(tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc)
```

```{r merge_rabund_peroccur, echo=F, results='hide'}
otu_tb16_ss36155_rabund_percoccur<-merge(tb16_tax_occur_ss36155_no_cero_t.rabund_means,tb16_tax_occur_ss36155_no_cero_t.rabund_means.occurence_perc, by="row.names")

colnames(otu_tb16_ss36155_rabund_percoccur)<-c("OTUs","mean_rabund","perc_occur")
otu_tb16_ss36155_rabund_percoccur[1:5,]

row.names(otu_tb16_ss36155_rabund_percoccur)<-otu_tb16_ss36155_rabund_percoccur[,1]
otu_tb16_ss36155_rabund_percoccur<-otu_tb16_ss36155_rabund_percoccur[,-1]
otu_tb16_ss36155_rabund_percoccur[1:5,]
```

In the following plot, we can appreciate the OTUs distribution according to their percentage of occurence and relative abundance. The red line keeps up OTUs that occur in more than 80% of the samples, the green line limits regionally rare OTUs (< 0.001%), and the blue one restricts regionally abundant OTUs (> 0.1%).

```{r abund_vs_occurence_table, echo=F}
plot(otu_tb16_ss36155_rabund_percoccur$mean_rabund,otu_tb16_ss36155_rabund_percoccur$perc_occur, log="x", pch=19, cex=0.8, xlab="Mean relative abundance", ylab="Percentage of occurence")
abline(h=80, col="red") #occurence higher than 80%
abline(v=0.00001, col="green") #rare OTUs
abline(v=0.001, col="blue") #cosmopolitan OTUs

#Conventional limits:
#Regionally rare     = 0.00001
#Regionally abundant = 0.001
```

Regionally abundant OTUs (relative abundance over 0.1%):

```{r abundant_OTUs, echo=7}
#regionally abundant
tb16_ss36155_abundant<-otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund > 0.001,]

tb16_ss36155_abundant_sorted<-tb16_ss36155_abundant[order(tb16_ss36155_abundant$mean_rabund, tb16_ss36155_abundant$perc_occur, decreasing = T), c(1,2)]

tb16_ss36155_abundant_sorted
dim(tb16_ss36155_abundant_sorted)
```

Proportion of regionally abundant OTUs (%):

```{r abundant_OTUs2, echo=F}
(nrow(tb16_ss36155_abundant_sorted)/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```

Cosmopolitan OTUs (relative abundance over 0.1% and occurence in more than 80% of samples):

```{r select_cosmopolitan, echo=6}
otu_tb16_ss36155_rabund_cosm<-otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund > 0.001,]
otu_tb16_ss36155_rabund_poccur_cosm<-otu_tb16_ss36155_rabund_cosm[otu_tb16_ss36155_rabund_cosm$perc_occur > 80,]
otu_tb16_ss36155_cosmop_sorted<-otu_tb16_ss36155_rabund_poccur_cosm[order(otu_tb16_ss36155_rabund_poccur_cosm$perc_occur, otu_tb16_ss36155_rabund_poccur_cosm$mean_rabund, decreasing = T), c(1,2)]

otu_tb16_ss36155_cosmop_sorted
dim(otu_tb16_ss36155_cosmop_sorted)
```

Number and proportion (%) of cosmopolitan OTUs:

```{r percentage_cosmopolitan, echo=F}
nrow(otu_tb16_ss36155_cosmop_sorted)

(nrow(otu_tb16_ss36155_cosmop_sorted)/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```

Number and proportion (%) of rare OTUs:

```{r rare_OTUs, echo=1}
nrow(otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund < 0.00001 & otu_tb16_ss36155_rabund_percoccur$mean_rabund >0,])

(nrow(otu_tb16_ss36155_rabund_percoccur[otu_tb16_ss36155_rabund_percoccur$mean_rabund < 0.00001 & otu_tb16_ss36155_rabund_percoccur$mean_rabund >0,])/nrow(otu_tb16_ss36155_rabund_percoccur))*100
```









